{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# STEAP"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook shows how the figures and tables were made."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/eerdem/STEAP'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# quick/dirty method to allow importing by changing working directory\n",
    "import os\n",
    "os.chdir(\"../\")\n",
    "os.getcwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import constants # all constants used here can be found and edited in the constants.py file\n",
    "import pandas as pd\n",
    "from pathlib import Path\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "from matplotlib_venn import venn3\n",
    "import matplotlib.patches as mpatches\n",
    "\n",
    "\n",
    "# from scripts import convert_output_to_dataframe\n",
    "from scripts import gene_set_enrichment_analysis\n",
    "# from scripts import calculate_beta_correlation\n",
    "# from scripts import calculate_es_correlation\n",
    "from scripts import upsetplot\n",
    "from scripts import circosplot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "figure_dir = 'figures_and_tables/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "methods = constants.METHODS\n",
    "pval_correction = constants.PVAL_CORRECTION\n",
    "gwas_group_dict = constants.GWAS_GROUP_DICT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gwas</th>\n",
       "      <th>specificity_id</th>\n",
       "      <th>annotation</th>\n",
       "      <th>beta</th>\n",
       "      <th>beta_se</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>method</th>\n",
       "      <th>n_methods</th>\n",
       "      <th>pvalue_bonferroni</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AD_JANSENS2019</td>\n",
       "      <td>Allen_human_LGN</td>\n",
       "      <td>GABAergic.LGN_Inh_CTXN3</td>\n",
       "      <td>-0.174645</td>\n",
       "      <td>0.089013</td>\n",
       "      <td>0.975106</td>\n",
       "      <td>H-MAGMA</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AD_JANSENS2019</td>\n",
       "      <td>Allen_human_LGN</td>\n",
       "      <td>GABAergic.LGN_Inh_LAMP5</td>\n",
       "      <td>-0.123540</td>\n",
       "      <td>0.071732</td>\n",
       "      <td>0.957472</td>\n",
       "      <td>H-MAGMA</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AD_JANSENS2019</td>\n",
       "      <td>Allen_human_LGN</td>\n",
       "      <td>GABAergic.LGN_Inh_NTRK1</td>\n",
       "      <td>-0.269440</td>\n",
       "      <td>0.088436</td>\n",
       "      <td>0.998840</td>\n",
       "      <td>H-MAGMA</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AD_JANSENS2019</td>\n",
       "      <td>Allen_human_LGN</td>\n",
       "      <td>GABAergic.LGN_Inh_TRPC4</td>\n",
       "      <td>-0.246886</td>\n",
       "      <td>0.099982</td>\n",
       "      <td>0.993223</td>\n",
       "      <td>H-MAGMA</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AD_JANSENS2019</td>\n",
       "      <td>Allen_human_LGN</td>\n",
       "      <td>Glutamatergic.LGN_Exc_BTNL9</td>\n",
       "      <td>-0.421298</td>\n",
       "      <td>0.133637</td>\n",
       "      <td>0.999188</td>\n",
       "      <td>H-MAGMA</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>879625</th>\n",
       "      <td>volume: thalamus</td>\n",
       "      <td>tabula_muris</td>\n",
       "      <td>Tongue.keratinocyte</td>\n",
       "      <td>-0.035636</td>\n",
       "      <td>0.045386</td>\n",
       "      <td>0.783817</td>\n",
       "      <td>MAGMA</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>879626</th>\n",
       "      <td>volume: thalamus</td>\n",
       "      <td>tabula_muris</td>\n",
       "      <td>Trachea.blood_cell</td>\n",
       "      <td>-0.102483</td>\n",
       "      <td>0.059198</td>\n",
       "      <td>0.958279</td>\n",
       "      <td>MAGMA</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>879627</th>\n",
       "      <td>volume: thalamus</td>\n",
       "      <td>tabula_muris</td>\n",
       "      <td>Trachea.endothelial_cell</td>\n",
       "      <td>-0.114722</td>\n",
       "      <td>0.055573</td>\n",
       "      <td>0.980498</td>\n",
       "      <td>MAGMA</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>879628</th>\n",
       "      <td>volume: thalamus</td>\n",
       "      <td>tabula_muris</td>\n",
       "      <td>Trachea.epithelial_cell</td>\n",
       "      <td>-0.102770</td>\n",
       "      <td>0.052743</td>\n",
       "      <td>0.974312</td>\n",
       "      <td>MAGMA</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>879629</th>\n",
       "      <td>volume: thalamus</td>\n",
       "      <td>tabula_muris</td>\n",
       "      <td>Trachea.mesenchymal_cell</td>\n",
       "      <td>0.050800</td>\n",
       "      <td>0.043447</td>\n",
       "      <td>0.121164</td>\n",
       "      <td>MAGMA</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>877995 rows Ã— 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    gwas   specificity_id                   annotation  \\\n",
       "0         AD_JANSENS2019  Allen_human_LGN      GABAergic.LGN_Inh_CTXN3   \n",
       "1         AD_JANSENS2019  Allen_human_LGN      GABAergic.LGN_Inh_LAMP5   \n",
       "2         AD_JANSENS2019  Allen_human_LGN      GABAergic.LGN_Inh_NTRK1   \n",
       "3         AD_JANSENS2019  Allen_human_LGN      GABAergic.LGN_Inh_TRPC4   \n",
       "4         AD_JANSENS2019  Allen_human_LGN  Glutamatergic.LGN_Exc_BTNL9   \n",
       "...                  ...              ...                          ...   \n",
       "879625  volume: thalamus     tabula_muris          Tongue.keratinocyte   \n",
       "879626  volume: thalamus     tabula_muris           Trachea.blood_cell   \n",
       "879627  volume: thalamus     tabula_muris     Trachea.endothelial_cell   \n",
       "879628  volume: thalamus     tabula_muris      Trachea.epithelial_cell   \n",
       "879629  volume: thalamus     tabula_muris     Trachea.mesenchymal_cell   \n",
       "\n",
       "            beta   beta_se    pvalue   method  n_methods  pvalue_bonferroni  \n",
       "0      -0.174645  0.089013  0.975106  H-MAGMA          3                1.0  \n",
       "1      -0.123540  0.071732  0.957472  H-MAGMA          3                1.0  \n",
       "2      -0.269440  0.088436  0.998840  H-MAGMA          3                1.0  \n",
       "3      -0.246886  0.099982  0.993223  H-MAGMA          3                1.0  \n",
       "4      -0.421298  0.133637  0.999188  H-MAGMA          3                1.0  \n",
       "...          ...       ...       ...      ...        ...                ...  \n",
       "879625 -0.035636  0.045386  0.783817    MAGMA          3                1.0  \n",
       "879626 -0.102483  0.059198  0.958279    MAGMA          3                1.0  \n",
       "879627 -0.114722  0.055573  0.980498    MAGMA          3                1.0  \n",
       "879628 -0.102770  0.052743  0.974312    MAGMA          3                1.0  \n",
       "879629  0.050800  0.043447  0.121164    MAGMA          3                1.0  \n",
       "\n",
       "[877995 rows x 9 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gwasx</th>\n",
       "      <th>gwasy</th>\n",
       "      <th>corr</th>\n",
       "      <th>pval</th>\n",
       "      <th>method</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AD_JANSENS2019</td>\n",
       "      <td>ASD_2019</td>\n",
       "      <td>-0.220065</td>\n",
       "      <td>2.278355e-07</td>\n",
       "      <td>H-MAGMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AD_JANSENS2019</td>\n",
       "      <td>BIP_PGC3</td>\n",
       "      <td>-0.122407</td>\n",
       "      <td>4.318468e-03</td>\n",
       "      <td>H-MAGMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AD_JANSENS2019</td>\n",
       "      <td>BMI_GIANT2018</td>\n",
       "      <td>-0.007513</td>\n",
       "      <td>8.614700e-01</td>\n",
       "      <td>H-MAGMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AD_JANSENS2019</td>\n",
       "      <td>DTI AD: ACR</td>\n",
       "      <td>0.247713</td>\n",
       "      <td>5.065423e-09</td>\n",
       "      <td>H-MAGMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AD_JANSENS2019</td>\n",
       "      <td>DTI AD: ALIC</td>\n",
       "      <td>0.234532</td>\n",
       "      <td>3.299649e-08</td>\n",
       "      <td>H-MAGMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144448</th>\n",
       "      <td>volume: caudate</td>\n",
       "      <td>volume: putamen</td>\n",
       "      <td>0.337825</td>\n",
       "      <td>6.184427e-16</td>\n",
       "      <td>MAGMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144449</th>\n",
       "      <td>volume: caudate</td>\n",
       "      <td>volume: thalamus</td>\n",
       "      <td>0.351890</td>\n",
       "      <td>3.040520e-17</td>\n",
       "      <td>MAGMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144450</th>\n",
       "      <td>volume: pallidum</td>\n",
       "      <td>volume: putamen</td>\n",
       "      <td>0.256756</td>\n",
       "      <td>1.313908e-09</td>\n",
       "      <td>MAGMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144451</th>\n",
       "      <td>volume: pallidum</td>\n",
       "      <td>volume: thalamus</td>\n",
       "      <td>0.310798</td>\n",
       "      <td>1.330849e-13</td>\n",
       "      <td>MAGMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144452</th>\n",
       "      <td>volume: putamen</td>\n",
       "      <td>volume: thalamus</td>\n",
       "      <td>0.229988</td>\n",
       "      <td>6.138297e-08</td>\n",
       "      <td>MAGMA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>431748 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   gwasx             gwasy      corr          pval   method\n",
       "0         AD_JANSENS2019          ASD_2019 -0.220065  2.278355e-07  H-MAGMA\n",
       "1         AD_JANSENS2019          BIP_PGC3 -0.122407  4.318468e-03  H-MAGMA\n",
       "2         AD_JANSENS2019     BMI_GIANT2018 -0.007513  8.614700e-01  H-MAGMA\n",
       "3         AD_JANSENS2019       DTI AD: ACR  0.247713  5.065423e-09  H-MAGMA\n",
       "4         AD_JANSENS2019      DTI AD: ALIC  0.234532  3.299649e-08  H-MAGMA\n",
       "...                  ...               ...       ...           ...      ...\n",
       "144448   volume: caudate   volume: putamen  0.337825  6.184427e-16    MAGMA\n",
       "144449   volume: caudate  volume: thalamus  0.351890  3.040520e-17    MAGMA\n",
       "144450  volume: pallidum   volume: putamen  0.256756  1.313908e-09    MAGMA\n",
       "144451  volume: pallidum  volume: thalamus  0.310798  1.330849e-13    MAGMA\n",
       "144452   volume: putamen  volume: thalamus  0.229988  6.138297e-08    MAGMA\n",
       "\n",
       "[431748 rows x 5 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>celltypex</th>\n",
       "      <th>celltypey</th>\n",
       "      <th>corr</th>\n",
       "      <th>pval</th>\n",
       "      <th>pval_bonferroni</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Allen_human_LGN, GABAergic.LGN_Inh_CTXN3</td>\n",
       "      <td>Allen_human_LGN, GABAergic.LGN_Inh_LAMP5</td>\n",
       "      <td>0.155456</td>\n",
       "      <td>3.503827e-15</td>\n",
       "      <td>5.194074e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Allen_human_LGN, GABAergic.LGN_Inh_CTXN3</td>\n",
       "      <td>Allen_human_LGN, GABAergic.LGN_Inh_NTRK1</td>\n",
       "      <td>0.360667</td>\n",
       "      <td>1.160120e-115</td>\n",
       "      <td>1.719762e-110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Allen_human_LGN, GABAergic.LGN_Inh_CTXN3</td>\n",
       "      <td>Allen_human_LGN, GABAergic.LGN_Inh_TRPC4</td>\n",
       "      <td>0.209286</td>\n",
       "      <td>6.473715e-26</td>\n",
       "      <td>9.596634e-21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Allen_human_LGN, GABAergic.LGN_Inh_CTXN3</td>\n",
       "      <td>Allen_human_LGN, Glutamatergic.LGN_Exc_BTNL9</td>\n",
       "      <td>0.159302</td>\n",
       "      <td>1.347408e-09</td>\n",
       "      <td>1.997397e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Allen_human_LGN, GABAergic.LGN_Inh_CTXN3</td>\n",
       "      <td>Allen_human_LGN, Glutamatergic.LGN_Exc_PRKCG_BCHE</td>\n",
       "      <td>0.246957</td>\n",
       "      <td>1.938275e-24</td>\n",
       "      <td>2.873299e-19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148235</th>\n",
       "      <td>tabula_muris, Trachea.blood_cell</td>\n",
       "      <td>tabula_muris, Trachea.epithelial_cell</td>\n",
       "      <td>0.079243</td>\n",
       "      <td>1.409471e-02</td>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148236</th>\n",
       "      <td>tabula_muris, Trachea.blood_cell</td>\n",
       "      <td>tabula_muris, Trachea.mesenchymal_cell</td>\n",
       "      <td>0.111152</td>\n",
       "      <td>1.375545e-07</td>\n",
       "      <td>2.039108e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148237</th>\n",
       "      <td>tabula_muris, Trachea.endothelial_cell</td>\n",
       "      <td>tabula_muris, Trachea.epithelial_cell</td>\n",
       "      <td>0.089872</td>\n",
       "      <td>9.832454e-04</td>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148238</th>\n",
       "      <td>tabula_muris, Trachea.endothelial_cell</td>\n",
       "      <td>tabula_muris, Trachea.mesenchymal_cell</td>\n",
       "      <td>0.149082</td>\n",
       "      <td>1.478933e-18</td>\n",
       "      <td>2.192370e-13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148239</th>\n",
       "      <td>tabula_muris, Trachea.epithelial_cell</td>\n",
       "      <td>tabula_muris, Trachea.mesenchymal_cell</td>\n",
       "      <td>-0.027608</td>\n",
       "      <td>1.459915e-01</td>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>148240 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       celltypex  \\\n",
       "0       Allen_human_LGN, GABAergic.LGN_Inh_CTXN3   \n",
       "1       Allen_human_LGN, GABAergic.LGN_Inh_CTXN3   \n",
       "2       Allen_human_LGN, GABAergic.LGN_Inh_CTXN3   \n",
       "3       Allen_human_LGN, GABAergic.LGN_Inh_CTXN3   \n",
       "4       Allen_human_LGN, GABAergic.LGN_Inh_CTXN3   \n",
       "...                                          ...   \n",
       "148235          tabula_muris, Trachea.blood_cell   \n",
       "148236          tabula_muris, Trachea.blood_cell   \n",
       "148237    tabula_muris, Trachea.endothelial_cell   \n",
       "148238    tabula_muris, Trachea.endothelial_cell   \n",
       "148239     tabula_muris, Trachea.epithelial_cell   \n",
       "\n",
       "                                                celltypey      corr  \\\n",
       "0                Allen_human_LGN, GABAergic.LGN_Inh_LAMP5  0.155456   \n",
       "1                Allen_human_LGN, GABAergic.LGN_Inh_NTRK1  0.360667   \n",
       "2                Allen_human_LGN, GABAergic.LGN_Inh_TRPC4  0.209286   \n",
       "3            Allen_human_LGN, Glutamatergic.LGN_Exc_BTNL9  0.159302   \n",
       "4       Allen_human_LGN, Glutamatergic.LGN_Exc_PRKCG_BCHE  0.246957   \n",
       "...                                                   ...       ...   \n",
       "148235              tabula_muris, Trachea.epithelial_cell  0.079243   \n",
       "148236             tabula_muris, Trachea.mesenchymal_cell  0.111152   \n",
       "148237              tabula_muris, Trachea.epithelial_cell  0.089872   \n",
       "148238             tabula_muris, Trachea.mesenchymal_cell  0.149082   \n",
       "148239             tabula_muris, Trachea.mesenchymal_cell -0.027608   \n",
       "\n",
       "                 pval  pval_bonferroni  \n",
       "0        3.503827e-15     5.194074e-10  \n",
       "1       1.160120e-115    1.719762e-110  \n",
       "2        6.473715e-26     9.596634e-21  \n",
       "3        1.347408e-09     1.997397e-04  \n",
       "4        1.938275e-24     2.873299e-19  \n",
       "...               ...              ...  \n",
       "148235   1.409471e-02     1.000000e+00  \n",
       "148236   1.375545e-07     2.039108e-02  \n",
       "148237   9.832454e-04     1.000000e+00  \n",
       "148238   1.478933e-18     2.192370e-13  \n",
       "148239   1.459915e-01     1.000000e+00  \n",
       "\n",
       "[148240 rows x 5 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df_all = pd.read_hdf('data/CELLECT_output/data.h5', 'df_all')\n",
    "corr_df = pd.read_hdf('data/CELLECT_output/data.h5', 'corr_df')\n",
    "es_corr_df = pd.read_hdf('data/CELLECT_output/data.h5', 'es_corr_df')\n",
    "display(df_all)\n",
    "display(corr_df)\n",
    "display(es_corr_df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Figure 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# venn diagram of shared significant annotations between methods\n",
    "plt.style.use('default')\n",
    "\n",
    "df_3m = df_all[(df_all['n_methods']==3)& # only get if analysed in all 3 methods\n",
    "               (df_all[f'pvalue_{pval_correction}']<=0.05)] # only significant\n",
    "df_3m = df_3m.groupby(['gwas','specificity_id','annotation'])['method'].agg(list).reset_index()\n",
    "df_3m = df_3m['method'].value_counts().reset_index()\n",
    "for m in methods:\n",
    "    df_3m[m] = df_3m['index'].apply(lambda x : m in x)\n",
    "df_3m.drop(columns='index', inplace=True)\n",
    "# this list needs this specific order for the venn subsets\n",
    "true_false_list = [\n",
    "    (True,False,False),\n",
    "    (False,True,False),\n",
    "    (True,True,False),\n",
    "    (False,False,True),\n",
    "    (True,False,True),\n",
    "    (False,True,True),\n",
    "    (True,True,True)\n",
    "]\n",
    "venn_values = []\n",
    "for a,b,c in true_false_list:\n",
    "    venn_values.append(\n",
    "        df_3m[(df_3m[methods[0]]==a)&(df_3m[methods[1]]==b)&(df_3m[methods[2]]==c)]['method'].values[0]\n",
    "    )\n",
    "\n",
    "v = venn3(subsets = venn_values, \n",
    "          set_labels = (methods))\n",
    "# change the colors...\n",
    "# v.get_patch_by_id('100').set_color('#4A148C')\n",
    "# v.get_patch_by_id('110').set_color('#9C27B0')\n",
    "# v.get_patch_by_id('010').set_color('#CE93D8')\n",
    "# v.get_patch_by_id('001').set_color('#9C27B0')\n",
    "# v.get_patch_by_id('011').set_color('#4A148C')\n",
    "# v.get_patch_by_id('111').set_color('#')\n",
    "plt.savefig(f\"{figure_dir}venn_diagram.png\", dpi=200, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Figure 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gwas_group_dict['Schizophrenia'] = ['SCZ_PGC3']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "upsetplot.plot_upset(df_all,\n",
    "                     gwas_group_dict,\n",
    "                     sign_threshold=len(methods)-1, # only use cell-types significant in >2 methods\n",
    "                     element_size=56,\n",
    "                     save=True, filename=f\"{figure_dir}upsetplot_main.png\")\n",
    "\n",
    "upsetplot.get_shared_celltypes(df_all,\n",
    "                               gwas_group_dict,\n",
    "                               sign_threshold=len(methods)-1,\n",
    "                               save_to_excel=True, filename=f\"{figure_dir}upsetplot_main.xlsx\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Figure 3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "mri_group_dict = {\n",
    "    'Volume' : ['volume'],\n",
    "    'Surface Area' : ['surface area'],\n",
    "    'Thickness' : ['thickness']\n",
    "}\n",
    "upsetplot.plot_upset(df_all, \n",
    "                     mri_group_dict,\n",
    "                     sign_threshold=len(methods)-1, # only use cell-types significant in >2 methods\n",
    "                     element_size=70,\n",
    "                     save=True, filename=f\"{figure_dir}upsetplot_structural_mri.png\"\n",
    "                    )\n",
    "\n",
    "upsetplot.get_shared_celltypes(df_all,\n",
    "                               mri_group_dict,\n",
    "                               sign_threshold=len(methods)-1,\n",
    "                               save_to_excel=True, filename=f\"{figure_dir}upsetplot_structural_mri.xlsx\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Figure 4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "dti_group_dict = {\n",
    "    'FA' : ['DTI FA'],\n",
    "    'MD' : ['DTI MD'],\n",
    "    'AD' : ['DTI AD'],\n",
    "    'RD' : ['DTI RD'],\n",
    "    'MO' : ['DTI MO'],\n",
    "#     'PC1' : ['DTI PC1'],\n",
    "#     'PC2' : ['DTI PC2'],\n",
    "#     'PC3' : ['DTI PC3'],\n",
    "#     'PC4' : ['DTI PC4'],\n",
    "#     'PC5' : ['DTI PC5'], \n",
    "}\n",
    "upsetplot.plot_upset(df_all, \n",
    "                     dti_group_dict,\n",
    "                     sign_threshold=len(methods)-1, # only use cell-types significant in >2 methods\n",
    "                     element_size=70, sort_by='degree',\n",
    "                     sort_categories_by=None, # None will sort based on the key order in the group_dict\n",
    "                     save=True, filename=f\"{figure_dir}upsetplot_dti.png\"\n",
    "                    )\n",
    "\n",
    "upsetplot.get_shared_celltypes(df_all,\n",
    "                               dti_group_dict,\n",
    "                               sign_threshold=len(methods)-1,\n",
    "                               save_to_excel=True, filename=f\"{figure_dir}upsetplot_dti.xlsx\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Figure 5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "dti_group_dict = {\n",
    "    'FA' : ['DTI FA'],\n",
    "    'MD' : ['DTI MD'],\n",
    "    'AD' : ['DTI AD'],\n",
    "    'RD' : ['DTI RD'],\n",
    "    'MO' : ['DTI MO'],\n",
    "    'PC1' : ['DTI PC1'],\n",
    "    'PC2' : ['DTI PC2'],\n",
    "    'PC3' : ['DTI PC3'],\n",
    "    'PC4' : ['DTI PC4'],\n",
    "    'PC5' : ['DTI PC5'], \n",
    "}\n",
    "upsetplot.plot_upset(df_all, \n",
    "                     dti_group_dict,\n",
    "                     sign_threshold=len(methods)-1, # only use cell-types significant in >2 methods\n",
    "                     element_size=70, sort_by='degree',\n",
    "                     sort_categories_by=None, # None will sort based on the key order in the group_dict\n",
    "                     save=True, filename=f\"{figure_dir}upsetplot_dti_w_pc.png\"\n",
    "                    )\n",
    "\n",
    "upsetplot.get_shared_celltypes(df_all,\n",
    "                               dti_group_dict,\n",
    "                               sign_threshold=len(methods)-1,\n",
    "                               save_to_excel=True, filename=f\"{figure_dir}upsetplot_dti_w_pc.xlsx\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Figure 6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "fmri_group_dict = {\n",
    "    'Node': ['Node'],\n",
    "    'Pair': ['Pair'],\n",
    "    'Edge': ['Edge']\n",
    "}\n",
    "upsetplot.plot_upset(df_all, \n",
    "                     fmri_group_dict,\n",
    "                     sign_threshold=len(methods)-1, # only use cell-types significant in >2 methods\n",
    "                     element_size=70, sort_by='degree',\n",
    "                     sort_categories_by=None, # None will sort based on the key order in the group_dict\n",
    "                     save=True, filename=f\"{figure_dir}upsetplot_fmri.png\"\n",
    "                    )\n",
    "\n",
    "upsetplot.get_shared_celltypes(df_all,\n",
    "                               fmri_group_dict,\n",
    "                               sign_threshold=len(methods)-1,\n",
    "                               save_to_excel=True, filename=f\"{figure_dir}upsetplot_fmri.xlsx\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Figure 7"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "gwas = \"SCZ_PGC3_2020\"\n",
    "gwas_name = \"Schizophrenia\"\n",
    "\n",
    "plt.style.use('seaborn')\n",
    "def explode_column(df, column, split_str=','):\n",
    "    df = df.set_index(\n",
    "            df.drop([column],axis=1).columns.to_list()\n",
    "        )[[column]].apply(lambda x: x.str.split(',').explode()).reset_index()\n",
    "    df[column] = df[column].str.strip()\n",
    "    return df\n",
    "\n",
    "ylab = [\"Class\", \"Developmental_compartment\", \"Neurotransmitter\",\"Region\"]\n",
    "heights = [3,3,4,9]\n",
    "\n",
    "#use mousebrain metadata\n",
    "file = f\"{figure_dir}mousebrain_celltype_metadata.csv\"\n",
    "mb_metadata_df = pd.read_csv(file)\n",
    "mb_metadata_df['Neurotransmitter'].replace(\"(\\w+) (\\(\\w+),\\s*(\\w+\\))\",r\"\\1 \\2), \\1 (\\3\",\n",
    "                                        regex=True, inplace=True) #to fix wrong explode\n",
    "mb_metadata_df['Type'] = 'All'\n",
    "df_gwas = df_all[(df_all['gwas']==gwas)\n",
    "                &\n",
    "                 (df_all['specificity_id']=='mousebrain')\n",
    "                 &\n",
    "                 (df_all[f\"pvalue_{pval_correction}\"]<=0.05)\n",
    "                ]\n",
    "\n",
    "df_gwas = df_gwas.groupby(['specificity_id','annotation']).size().reset_index().rename(columns={0:'N_methods'})\n",
    "annot_list = df_gwas[df_gwas['N_methods']>=len(methods)-1].annotation.to_list()\n",
    "mb_metadata_df.loc[mb_metadata_df.ClusterName.isin(annot_list), 'Type'] = gwas\n",
    "\n",
    "fig = plt.figure(figsize=(8,8))\n",
    "ax_position = [321,323,325,122]\n",
    "for pos,y in zip(ax_position,ylab):\n",
    "    ax = fig.add_subplot(pos)\n",
    "    df = explode_column(mb_metadata_df,y)\n",
    "    df[[y,'Type']].groupby([y,'Type']).size().reset_index()\\\n",
    "    .pivot_table(values=0, index=y, columns='Type').sort_values(y, ascending=False).plot.barh(\n",
    "        stacked=True, legend=False, ax=ax)\n",
    "    ax.set_ylabel('')\n",
    "    ax.set_title(y.replace('_',' '))\n",
    "# plt.suptitle(gwas)\n",
    "plt.legend(['__',f\"{gwas_name}\"])\n",
    "fig.text(0.5, -0.0, 'Count', ha='center')\n",
    "plt.tight_layout()\n",
    "plt.savefig(f\"{figure_dir}mousebrain_{gwas_name}.png\", dpi=200, bbox_inches='tight')\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Figure 8"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Performing GSEA...\n",
      "\n",
      "Top None ranked cell-types (N=17) in Structural_MRI GWAS:\n",
      "1. GSE67835_Human_Cortex: fetal_quiescent (N=5, freq=0.044)\n",
      "2. GSE67835_Human_Cortex: endothelial (N=3, freq=0.027)\n",
      "2. DroNc_Human_Hippocampus: ODC1 (N=3, freq=0.027)\n",
      "3. Allen_human_LGN: Non-Neuronal.Oligo_MAG (N=2, freq=0.018)\n",
      "3. mousebrain: CBNBL2 (N=2, freq=0.018)\n",
      "3. Descartes_Human_Cerebrum: Cerebrum-Excitatory_neurons (N=2, freq=0.018)\n",
      "3. GSE67835_Human_Cortex: fetal_replicating (N=2, freq=0.018)\n",
      "4. PsychENCODE_DER-22: Ex9 (N=1, freq=0.0088)\n",
      "4. GSE67835_Human_Cortex: oligodendrocytes (N=1, freq=0.0088)\n",
      "4. DroNc_Human_Hippocampus: exPFC2 (N=1, freq=0.0088)\n",
      "4. GSE101601_Temporal_Cortex: gaba_sst (N=1, freq=0.0088)\n",
      "4. Allen_human_MTG: GABAergic.Inh_L5-6_SST_TH (N=1, freq=0.0088)\n",
      "4. DroNc_Human_Hippocampus: exCA3 (N=1, freq=0.0088)\n",
      "4. DroNc_Human_Hippocampus: NSC (N=1, freq=0.0088)\n",
      "4. Descartes_Human_Cerebrum: Cerebrum-Megakaryocytes (N=1, freq=0.0088)\n",
      "4. Descartes_Human_Cerebellum: Cerebellum-Unipolar_brush_cells (N=1, freq=0.0088)\n",
      "4. mousebrain: VLMC2 (N=1, freq=0.0088)\n",
      "\n",
      "\n",
      "Converting Ensembl ID to Gene Symbol...\n",
      "\n",
      "(1/17) Converting GSE67835_Human_Cortex, fetal_quiescent (39 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(2/17) Converting GSE67835_Human_Cortex, endothelial (39 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(3/17) Converting DroNc_Human_Hippocampus, ODC1 (124 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(4/17) Converting Allen_human_LGN, Non-Neuronal.Oligo_MAG (125 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(5/17) Converting mousebrain, CBNBL2 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(6/17) Converting Descartes_Human_Cerebrum, Cerebrum-Excitatory_neurons (152 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(7/17) Converting GSE67835_Human_Cortex, fetal_replicating (39 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(8/17) Converting PsychENCODE_DER-22, Ex9 (146 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(9/17) Converting GSE67835_Human_Cortex, oligodendrocytes (39 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(10/17) Converting DroNc_Human_Hippocampus, exPFC2 (124 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(11/17) Converting GSE101601_Temporal_Cortex, gaba_sst (16 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(12/17) Converting Allen_human_MTG, GABAergic.Inh_L5-6_SST_TH (181 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(13/17) Converting DroNc_Human_Hippocampus, exCA3 (124 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(14/17) Converting DroNc_Human_Hippocampus, NSC (124 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(15/17) Converting Descartes_Human_Cerebrum, Cerebrum-Megakaryocytes (152 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(16/17) Converting Descartes_Human_Cerebellum, Cerebellum-Unipolar_brush_cells (71 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(17/17) Converting mousebrain, VLMC2 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "Running Enrichr...\n",
      "\n",
      "(1/17) Analyzing GSE67835_Human_Cortex, fetal_quiescent (39 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(2/17) Analyzing GSE67835_Human_Cortex, endothelial (39 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(3/17) Analyzing DroNc_Human_Hippocampus, ODC1 (124 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(4/17) Analyzing Allen_human_LGN, Non-Neuronal.Oligo_MAG (125 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(5/17) Analyzing mousebrain, CBNBL2 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(6/17) Analyzing Descartes_Human_Cerebrum, Cerebrum-Excitatory_neurons (152 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(7/17) Analyzing GSE67835_Human_Cortex, fetal_replicating (39 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(8/17) Analyzing PsychENCODE_DER-22, Ex9 (146 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(9/17) Analyzing GSE67835_Human_Cortex, oligodendrocytes (39 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(10/17) Analyzing DroNc_Human_Hippocampus, exPFC2 (124 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(11/17) Analyzing GSE101601_Temporal_Cortex, gaba_sst (16 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(12/17) Analyzing Allen_human_MTG, GABAergic.Inh_L5-6_SST_TH (181 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(13/17) Analyzing DroNc_Human_Hippocampus, exCA3 (124 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(14/17) Analyzing DroNc_Human_Hippocampus, NSC (124 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(15/17) Analyzing Descartes_Human_Cerebrum, Cerebrum-Megakaryocytes (152 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(16/17) Analyzing Descartes_Human_Cerebellum, Cerebellum-Unipolar_brush_cells (71 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(17/17) Analyzing mousebrain, VLMC2 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "Performing GSEA...\n",
      "\n",
      "Top None ranked cell-types (N=49) in DTI_tracts GWAS:\n",
      "1. GSE67835_Human_Cortex: fetal_quiescent (N=26, freq=0.12)\n",
      "2. Allen_human_LGN: Non-Neuronal.Oligo_MAG (N=13, freq=0.06)\n",
      "2. DroNc_Human_Hippocampus: ODC1 (N=13, freq=0.06)\n",
      "2. tabula_muris: Marrow.common_lymphoid_progenitor (N=13, freq=0.06)\n",
      "3. tabula_muris: Marrow.late_pro-B_cell (N=12, freq=0.056)\n",
      "4. Descartes_Human_Cerebrum: Cerebrum-Vascular_endothelial_cells (N=11, freq=0.051)\n",
      "5. mousebrain: MFOL1 (N=10, freq=0.047)\n",
      "5. mousebrain: MFOL2 (N=10, freq=0.047)\n",
      "6. DroNc_Human_Hippocampus: ODC2 (N=9, freq=0.042)\n",
      "6. Descartes_Human_Cerebrum: Cerebrum-Excitatory_neurons (N=9, freq=0.042)\n",
      "7. Allen_human_LGN: Non-Neuronal.Astro_LGR6 (N=8, freq=0.037)\n",
      "8. PsychENCODE_DER-22: Oligo (N=7, freq=0.033)\n",
      "9. GSE67835_Human_Cortex: endothelial (N=6, freq=0.028)\n",
      "10. Allen_human_MTG: Non-neuronal.Oligo_L1-6_OPALIN (N=5, freq=0.023)\n",
      "11. Descartes_Human_Cerebellum: Cerebellum-Astrocytes (N=3, freq=0.014)\n",
      "11. tabula_muris: Brain_Non-Myeloid.oligodendrocyte (N=3, freq=0.014)\n",
      "11. DroNc_Human_Hippocampus: exCA1 (N=3, freq=0.014)\n",
      "11. DroNc_Human_Hippocampus: exCA3 (N=3, freq=0.014)\n",
      "11. GSE101601_Temporal_Cortex: gaba_fbxl7 (N=3, freq=0.014)\n",
      "11. DroNc_Human_Hippocampus: ASC2 (N=3, freq=0.014)\n",
      "11. tabula_muris: Marrow.immature_B_cell (N=3, freq=0.014)\n",
      "12. tabula_muris: Marrow.granulocytopoietic_cell (N=2, freq=0.0093)\n",
      "12. mousebrain: HBGLU5 (N=2, freq=0.0093)\n",
      "12. mousebrain: MOL2 (N=2, freq=0.0093)\n",
      "12. mousebrain: MOL1 (N=2, freq=0.0093)\n",
      "12. tabula_muris: Pancreas.type_B_pancreatic_cell (N=2, freq=0.0093)\n",
      "12. mousebrain: VLMC2 (N=2, freq=0.0093)\n",
      "12. PsychENCODE_DER-22: Ex3e (N=2, freq=0.0093)\n",
      "12. PsychENCODE_DER-22: Per (N=2, freq=0.0093)\n",
      "12. Allen_human_MTG: Non-neuronal.Astro_L1-2_FGFR3_GFAP (N=2, freq=0.0093)\n",
      "12. PsychENCODE_DER-22: Astro (N=2, freq=0.0093)\n",
      "13. tabula_muris: Limb_Muscle.endothelial_cell (N=1, freq=0.0047)\n",
      "13. Allen_human_MTG: GABAergic.Inh_L5-6_SST_KLHDC8A (N=1, freq=0.0047)\n",
      "13. Allen_human_MTG: Non-neuronal.Astro_L1-6_FGFR3_SLC14A1 (N=1, freq=0.0047)\n",
      "13. Descartes_Human_Cerebrum: Cerebrum-Astrocytes (N=1, freq=0.0047)\n",
      "13. DroNc_Human_Hippocampus: NSC (N=1, freq=0.0047)\n",
      "13. tabula_muris: Marrow.basophil (N=1, freq=0.0047)\n",
      "13. tabula_muris: Marrow.Slamf1-positive_multipotent_progenitor_cell (N=1, freq=0.0047)\n",
      "13. tabula_muris: Limb_Muscle.skeletal_muscle_satellite_cell (N=1, freq=0.0047)\n",
      "13. tabula_muris: Fat.mesenchymal_stem_cell_of_adipose (N=1, freq=0.0047)\n",
      "13. DroNc_Human_Hippocampus: OPC (N=1, freq=0.0047)\n",
      "13. tabula_muris: Brain_Non-Myeloid.endothelial_cell (N=1, freq=0.0047)\n",
      "13. GSE101601_Temporal_Cortex: gaba_reln (N=1, freq=0.0047)\n",
      "13. mousebrain: VLMC1 (N=1, freq=0.0047)\n",
      "13. mousebrain: VECA (N=1, freq=0.0047)\n",
      "13. mousebrain: NFOL2 (N=1, freq=0.0047)\n",
      "13. GSE67835_Human_Cortex: fetal_replicating (N=1, freq=0.0047)\n",
      "13. GSE67835_Human_Cortex: oligodendrocytes (N=1, freq=0.0047)\n",
      "13. tabula_muris: Trachea.endothelial_cell (N=1, freq=0.0047)\n",
      "\n",
      "\n",
      "Converting Ensembl ID to Gene Symbol...\n",
      "\n",
      "(1/49) Converting GSE67835_Human_Cortex, fetal_quiescent (39 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(2/49) Converting Allen_human_LGN, Non-Neuronal.Oligo_MAG (125 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(3/49) Converting DroNc_Human_Hippocampus, ODC1 (124 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(4/49) Converting tabula_muris, Marrow.common_lymphoid_progenitor (143 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(5/49) Converting tabula_muris, Marrow.late_pro-B_cell (143 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(6/49) Converting Descartes_Human_Cerebrum, Cerebrum-Vascular_endothelial_cells (152 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(7/49) Converting mousebrain, MFOL1 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(8/49) Converting mousebrain, MFOL2 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(9/49) Converting DroNc_Human_Hippocampus, ODC2 (124 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(10/49) Converting Descartes_Human_Cerebrum, Cerebrum-Excitatory_neurons (152 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(11/49) Converting Allen_human_LGN, Non-Neuronal.Astro_LGR6 (125 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(12/49) Converting PsychENCODE_DER-22, Oligo (146 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(13/49) Converting GSE67835_Human_Cortex, endothelial (39 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(14/49) Converting Allen_human_MTG, Non-neuronal.Oligo_L1-6_OPALIN (181 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(15/49) Converting Descartes_Human_Cerebellum, Cerebellum-Astrocytes (71 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(16/49) Converting tabula_muris, Brain_Non-Myeloid.oligodendrocyte (143 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(17/49) Converting DroNc_Human_Hippocampus, exCA1 (124 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(18/49) Converting DroNc_Human_Hippocampus, exCA3 (124 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(19/49) Converting GSE101601_Temporal_Cortex, gaba_fbxl7 (16 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(20/49) Converting DroNc_Human_Hippocampus, ASC2 (124 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(21/49) Converting tabula_muris, Marrow.immature_B_cell (143 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(22/49) Converting tabula_muris, Marrow.granulocytopoietic_cell (143 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(23/49) Converting mousebrain, HBGLU5 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(24/49) Converting mousebrain, MOL2 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(25/49) Converting mousebrain, MOL1 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(26/49) Converting tabula_muris, Pancreas.type_B_pancreatic_cell (143 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(27/49) Converting mousebrain, VLMC2 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(28/49) Converting PsychENCODE_DER-22, Ex3e (146 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(29/49) Converting PsychENCODE_DER-22, Per (146 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(30/49) Converting Allen_human_MTG, Non-neuronal.Astro_L1-2_FGFR3_GFAP (181 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(31/49) Converting PsychENCODE_DER-22, Astro (146 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(32/49) Converting tabula_muris, Limb_Muscle.endothelial_cell (143 genes)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "querying 1-143...done.\n",
      "Finished.\n",
      "2 input query terms found no hit:\n",
      "\t['ENSG00000182851', 'ENSG00000143140']\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "(33/49) Converting Allen_human_MTG, GABAergic.Inh_L5-6_SST_KLHDC8A (181 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(34/49) Converting Allen_human_MTG, Non-neuronal.Astro_L1-6_FGFR3_SLC14A1 (181 genes)\n",
      "querying 1-181...done.\n",
      "Finished.\n",
      "\n",
      "(35/49) Converting Descartes_Human_Cerebrum, Cerebrum-Astrocytes (152 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(36/49) Converting DroNc_Human_Hippocampus, NSC (124 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(37/49) Converting tabula_muris, Marrow.basophil (143 genes)\n",
      "querying 1-143...done.\n",
      "Finished.\n",
      "4 input query terms found no hit:\n",
      "\t['ENSG00000197697', 'ENSG00000196226', 'ENSG00000203818', 'ENSG00000184348']\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "(38/49) Converting tabula_muris, Marrow.Slamf1-positive_multipotent_progenitor_cell (143 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(39/49) Converting tabula_muris, Limb_Muscle.skeletal_muscle_satellite_cell (143 genes)\n",
      "querying 1-143...done.\n",
      "Finished.\n",
      "\n",
      "(40/49) Converting tabula_muris, Fat.mesenchymal_stem_cell_of_adipose (143 genes)\n",
      "querying 1-143...done.\n",
      "Finished.\n",
      "1 input query terms found no hit:\n",
      "\t['ENSG00000107623']\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "(41/49) Converting DroNc_Human_Hippocampus, OPC (124 genes)\n",
      "querying 1-124...done.\n",
      "Finished.\n",
      "4 input query terms found no hit:\n",
      "\t['ENSG00000263254', 'ENSG00000268760', 'ENSG00000155130', 'ENSG00000272641']\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "(42/49) Converting tabula_muris, Brain_Non-Myeloid.endothelial_cell (143 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(43/49) Converting GSE101601_Temporal_Cortex, gaba_reln (16 genes)\n",
      "querying 1-16...done.\n",
      "Finished.\n",
      "\n",
      "(44/49) Converting mousebrain, VLMC1 (151 genes)\n",
      "querying 1-151...done.\n",
      "Finished.\n",
      "\n",
      "(45/49) Converting mousebrain, VECA (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(46/49) Converting mousebrain, NFOL2 (151 genes)\n",
      "querying 1-151...done.\n",
      "Finished.\n",
      "1 input query terms found no hit:\n",
      "\t['ENSG00000165388']\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "(47/49) Converting GSE67835_Human_Cortex, fetal_replicating (39 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(48/49) Converting GSE67835_Human_Cortex, oligodendrocytes (39 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(49/49) Converting tabula_muris, Trachea.endothelial_cell (143 genes)\n",
      "querying 1-143...done.\n",
      "Finished.\n",
      "\n",
      "Running Enrichr...\n",
      "\n",
      "(1/49) Analyzing GSE67835_Human_Cortex, fetal_quiescent (39 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(2/49) Analyzing Allen_human_LGN, Non-Neuronal.Oligo_MAG (125 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(3/49) Analyzing DroNc_Human_Hippocampus, ODC1 (124 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(4/49) Analyzing tabula_muris, Marrow.common_lymphoid_progenitor (143 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(5/49) Analyzing tabula_muris, Marrow.late_pro-B_cell (143 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(6/49) Analyzing Descartes_Human_Cerebrum, Cerebrum-Vascular_endothelial_cells (152 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(7/49) Analyzing mousebrain, MFOL1 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(8/49) Analyzing mousebrain, MFOL2 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(9/49) Analyzing DroNc_Human_Hippocampus, ODC2 (124 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(10/49) Analyzing Descartes_Human_Cerebrum, Cerebrum-Excitatory_neurons (152 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(11/49) Analyzing Allen_human_LGN, Non-Neuronal.Astro_LGR6 (125 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(12/49) Analyzing PsychENCODE_DER-22, Oligo (146 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(13/49) Analyzing GSE67835_Human_Cortex, endothelial (39 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(14/49) Analyzing Allen_human_MTG, Non-neuronal.Oligo_L1-6_OPALIN (181 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(15/49) Analyzing Descartes_Human_Cerebellum, Cerebellum-Astrocytes (71 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(16/49) Analyzing tabula_muris, Brain_Non-Myeloid.oligodendrocyte (143 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(17/49) Analyzing DroNc_Human_Hippocampus, exCA1 (124 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(18/49) Analyzing DroNc_Human_Hippocampus, exCA3 (124 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(19/49) Analyzing GSE101601_Temporal_Cortex, gaba_fbxl7 (16 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(20/49) Analyzing DroNc_Human_Hippocampus, ASC2 (124 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(21/49) Analyzing tabula_muris, Marrow.immature_B_cell (143 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(22/49) Analyzing tabula_muris, Marrow.granulocytopoietic_cell (143 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(23/49) Analyzing mousebrain, HBGLU5 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(24/49) Analyzing mousebrain, MOL2 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(25/49) Analyzing mousebrain, MOL1 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(26/49) Analyzing tabula_muris, Pancreas.type_B_pancreatic_cell (143 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(27/49) Analyzing mousebrain, VLMC2 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(28/49) Analyzing PsychENCODE_DER-22, Ex3e (146 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(29/49) Analyzing PsychENCODE_DER-22, Per (146 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(30/49) Analyzing Allen_human_MTG, Non-neuronal.Astro_L1-2_FGFR3_GFAP (181 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(31/49) Analyzing PsychENCODE_DER-22, Astro (146 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(32/49) Analyzing tabula_muris, Limb_Muscle.endothelial_cell (141 genes)...\n",
      "Writing to file...\n",
      "\n",
      "(33/49) Analyzing Allen_human_MTG, GABAergic.Inh_L5-6_SST_KLHDC8A (181 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(34/49) Analyzing Allen_human_MTG, Non-neuronal.Astro_L1-6_FGFR3_SLC14A1 (181 genes)...\n",
      "Writing to file...\n",
      "\n",
      "(35/49) Analyzing Descartes_Human_Cerebrum, Cerebrum-Astrocytes (152 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(36/49) Analyzing DroNc_Human_Hippocampus, NSC (124 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(37/49) Analyzing tabula_muris, Marrow.basophil (139 genes)...\n",
      "Writing to file...\n",
      "\n",
      "(38/49) Analyzing tabula_muris, Marrow.Slamf1-positive_multipotent_progenitor_cell (143 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(39/49) Analyzing tabula_muris, Limb_Muscle.skeletal_muscle_satellite_cell (143 genes)...\n",
      "Writing to file...\n",
      "\n",
      "(40/49) Analyzing tabula_muris, Fat.mesenchymal_stem_cell_of_adipose (142 genes)...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-03-23 13:43:12,270 Error fetching enrichment results: Epigenomics_Roadmap_HM_ChIP-seq\n",
      "2021-03-23 13:43:12,462 Error fetching enrichment results: Epigenomics_Roadmap_HM_ChIP-seq\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Expecting value: line 1 column 1 (char 0)\n",
      "Writing to file...\n",
      "\n",
      "(41/49) Analyzing DroNc_Human_Hippocampus, OPC (120 genes)...\n",
      "Writing to file...\n",
      "\n",
      "(42/49) Analyzing tabula_muris, Brain_Non-Myeloid.endothelial_cell (143 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(43/49) Analyzing GSE101601_Temporal_Cortex, gaba_reln (16 genes)...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-03-23 13:46:39,189 Error fetching enrichment results: Tissue_Protein_Expression_from_ProteomicsDB\n",
      "2021-03-23 13:46:39,381 Error fetching enrichment results: Tissue_Protein_Expression_from_ProteomicsDB\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Expecting value: line 1 column 1 (char 0)\n",
      "Writing to file...\n",
      "\n",
      "(44/49) Analyzing mousebrain, VLMC1 (151 genes)...\n",
      "Writing to file...\n",
      "\n",
      "(45/49) Analyzing mousebrain, VECA (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(46/49) Analyzing mousebrain, NFOL2 (150 genes)...\n",
      "Writing to file...\n",
      "\n",
      "(47/49) Analyzing GSE67835_Human_Cortex, fetal_replicating (39 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(48/49) Analyzing GSE67835_Human_Cortex, oligodendrocytes (39 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(49/49) Analyzing tabula_muris, Trachea.endothelial_cell (143 genes)...\n",
      "Writing to file...\n",
      "Performing GSEA...\n",
      "\n",
      "Top None ranked cell-types (N=40) in rsfMRI GWAS:\n",
      "1. tabula_muris: Brain_Non-Myeloid.brain_pericyte (N=33, freq=0.17)\n",
      "2. Descartes_Human_Cerebrum: Cerebrum-Vascular_endothelial_cells (N=14, freq=0.07)\n",
      "3. Descartes_Human_Cerebellum: Cerebellum-Vascular_endothelial_cells (N=10, freq=0.05)\n",
      "4. Descartes_Human_Cerebrum: Cerebrum-Excitatory_neurons (N=9, freq=0.045)\n",
      "5. tabula_muris: Heart.smooth_muscle_cell (N=7, freq=0.035)\n",
      "6. GSE67835_Human_Cortex: fetal_quiescent (N=6, freq=0.03)\n",
      "7. tabula_muris: Heart.myofibroblast_cell (N=4, freq=0.02)\n",
      "8. tabula_muris: Brain_Non-Myeloid.endothelial_cell (N=3, freq=0.015)\n",
      "8. GSE67835_Human_Cortex: endothelial (N=3, freq=0.015)\n",
      "9. Descartes_Human_Cerebrum: Cerebrum-Inhibitory_neurons (N=2, freq=0.01)\n",
      "9. Descartes_Human_Cerebellum: Cerebellum-Inhibitory_interneurons (N=2, freq=0.01)\n",
      "9. tabula_muris: Marrow.late_pro-B_cell (N=2, freq=0.01)\n",
      "10. PsychENCODE_DER-22: Ex8 (N=1, freq=0.005)\n",
      "10. tabula_muris: Fat.mesenchymal_stem_cell_of_adipose (N=1, freq=0.005)\n",
      "10. tabula_muris: Marrow.granulocytopoietic_cell (N=1, freq=0.005)\n",
      "10. mousebrain: VECA (N=1, freq=0.005)\n",
      "10. mousebrain: SCINH5 (N=1, freq=0.005)\n",
      "10. mousebrain: SCINH2 (N=1, freq=0.005)\n",
      "10. mousebrain: SCINH10 (N=1, freq=0.005)\n",
      "10. mousebrain: HBINH8 (N=1, freq=0.005)\n",
      "10. mousebrain: ENTG1 (N=1, freq=0.005)\n",
      "10. mousebrain: DEGLU4 (N=1, freq=0.005)\n",
      "10. PsychENCODE_DER-22: Oligo (N=1, freq=0.005)\n",
      "10. Allen_human_LGN: GABAergic.LGN_Inh_LAMP5 (N=1, freq=0.005)\n",
      "10. PsychENCODE_DER-22: Ex6a (N=1, freq=0.005)\n",
      "10. GSE67835_Human_Cortex: oligodendrocytes (N=1, freq=0.005)\n",
      "10. Allen_human_LGN: Glutamatergic.LGN_Exc_BTNL9 (N=1, freq=0.005)\n",
      "10. GSE101601_Temporal_Cortex: htrc2 (N=1, freq=0.005)\n",
      "10. DroNc_Human_Hippocampus: exPFC1 (N=1, freq=0.005)\n",
      "10. DroNc_Human_Hippocampus: exDG (N=1, freq=0.005)\n",
      "10. DroNc_Human_Hippocampus: exCA3 (N=1, freq=0.005)\n",
      "10. DroNc_Human_Hippocampus: MG (N=1, freq=0.005)\n",
      "10. DroNc_Human_Hippocampus: END (N=1, freq=0.005)\n",
      "10. Descartes_Human_Cerebrum: Cerebrum-Astrocytes (N=1, freq=0.005)\n",
      "10. Allen_human_MTG: Glutamatergic.Exc_L5-6_THEMIS_FGF10 (N=1, freq=0.005)\n",
      "10. Allen_human_MTG: Glutamatergic.Exc_L5-6_THEMIS_DCSTAMP (N=1, freq=0.005)\n",
      "10. Allen_human_MTG: Glutamatergic.Exc_L5-6_THEMIS_CRABP1 (N=1, freq=0.005)\n",
      "10. Allen_human_MTG: GABAergic.Inh_L1_SST_NMBR (N=1, freq=0.005)\n",
      "10. Allen_human_MTG: GABAergic.Inh_L1-2_PAX6_CDH12 (N=1, freq=0.005)\n",
      "10. tabula_muris: Marrow.precursor_B_cell (N=1, freq=0.005)\n",
      "\n",
      "\n",
      "Converting Ensembl ID to Gene Symbol...\n",
      "\n",
      "(1/40) Converting tabula_muris, Brain_Non-Myeloid.brain_pericyte (143 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(2/40) Converting Descartes_Human_Cerebrum, Cerebrum-Vascular_endothelial_cells (152 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(3/40) Converting Descartes_Human_Cerebellum, Cerebellum-Vascular_endothelial_cells (71 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(4/40) Converting Descartes_Human_Cerebrum, Cerebrum-Excitatory_neurons (152 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(5/40) Converting tabula_muris, Heart.smooth_muscle_cell (143 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(6/40) Converting GSE67835_Human_Cortex, fetal_quiescent (39 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(7/40) Converting tabula_muris, Heart.myofibroblast_cell (143 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(8/40) Converting tabula_muris, Brain_Non-Myeloid.endothelial_cell (143 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(9/40) Converting GSE67835_Human_Cortex, endothelial (39 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(10/40) Converting Descartes_Human_Cerebrum, Cerebrum-Inhibitory_neurons (152 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(11/40) Converting Descartes_Human_Cerebellum, Cerebellum-Inhibitory_interneurons (71 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(12/40) Converting tabula_muris, Marrow.late_pro-B_cell (143 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(13/40) Converting PsychENCODE_DER-22, Ex8 (146 genes)\n",
      "querying 1-146...done.\n",
      "Finished.\n",
      "\n",
      "(14/40) Converting tabula_muris, Fat.mesenchymal_stem_cell_of_adipose (143 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(15/40) Converting tabula_muris, Marrow.granulocytopoietic_cell (143 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(16/40) Converting mousebrain, VECA (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(17/40) Converting mousebrain, SCINH5 (151 genes)\n",
      "querying 1-151...done.\n",
      "Finished.\n",
      "3 input query terms found no hit:\n",
      "\t['ENSG00000128802', 'ENSG00000165862', 'ENSG00000132130']\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "(18/40) Converting mousebrain, SCINH2 (151 genes)\n",
      "querying 1-151...done.\n",
      "Finished.\n",
      "4 input query terms found no hit:\n",
      "\t['ENSG00000132130', 'ENSG00000249034', 'ENSG00000214732', 'ENSG00000133808']\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "(19/40) Converting mousebrain, SCINH10 (151 genes)\n",
      "querying 1-151...done.\n",
      "Finished.\n",
      "2 input query terms found no hit:\n",
      "\t['ENSG00000233224', 'ENSG00000182217']\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "(20/40) Converting mousebrain, HBINH8 (151 genes)\n",
      "querying 1-151...done.\n",
      "Finished.\n",
      "2 input query terms found no hit:\n",
      "\t['ENSG00000214556', 'ENSG00000269071']\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "(21/40) Converting mousebrain, ENTG1 (151 genes)\n",
      "querying 1-151...done.\n",
      "Finished.\n",
      "4 input query terms found no hit:\n",
      "\t['ENSG00000198374', 'ENSG00000168078', 'ENSG00000182611', 'ENSG00000233224']\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "(22/40) Converting mousebrain, DEGLU4 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(23/40) Converting PsychENCODE_DER-22, Oligo (146 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(24/40) Converting Allen_human_LGN, GABAergic.LGN_Inh_LAMP5 (125 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(25/40) Converting PsychENCODE_DER-22, Ex6a (146 genes)\n",
      "querying 1-146...done.\n",
      "Finished.\n",
      "\n",
      "(26/40) Converting GSE67835_Human_Cortex, oligodendrocytes (39 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(27/40) Converting Allen_human_LGN, Glutamatergic.LGN_Exc_BTNL9 (125 genes)\n",
      "querying 1-125...done.\n",
      "Finished.\n",
      "\n",
      "(28/40) Converting GSE101601_Temporal_Cortex, htrc2 (16 genes)\n",
      "querying 1-16...done.\n",
      "Finished.\n",
      "\n",
      "(29/40) Converting DroNc_Human_Hippocampus, exPFC1 (124 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(30/40) Converting DroNc_Human_Hippocampus, exDG (124 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(31/40) Converting DroNc_Human_Hippocampus, exCA3 (124 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(32/40) Converting DroNc_Human_Hippocampus, MG (124 genes)\n",
      "querying 1-124...done.\n",
      "Finished.\n",
      "12 input query terms found no hit:\n",
      "\t['ENSG00000266332', 'ENSG00000261919', 'ENSG00000262397', 'ENSG00000263850', 'ENSG00000168274', 'ENS\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "(33/40) Converting DroNc_Human_Hippocampus, END (124 genes)\n",
      "querying 1-124...done.\n",
      "Finished.\n",
      "\n",
      "(34/40) Converting Descartes_Human_Cerebrum, Cerebrum-Astrocytes (152 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(35/40) Converting Allen_human_MTG, Glutamatergic.Exc_L5-6_THEMIS_FGF10 (181 genes)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "querying 1-181...done.\n",
      "Finished.\n",
      "2 input query terms found no hit:\n",
      "\t['ENSG00000278272', 'ENSG00000250889']\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "(36/40) Converting Allen_human_MTG, Glutamatergic.Exc_L5-6_THEMIS_DCSTAMP (181 genes)\n",
      "querying 1-181...done.\n",
      "Finished.\n",
      "2 input query terms found no hit:\n",
      "\t['ENSG00000241978', 'ENSG00000250889']\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "(37/40) Converting Allen_human_MTG, Glutamatergic.Exc_L5-6_THEMIS_CRABP1 (181 genes)\n",
      "querying 1-181...done.\n",
      "Finished.\n",
      "2 input query terms found no hit:\n",
      "\t['ENSG00000250889', 'ENSG00000225986']\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "(38/40) Converting Allen_human_MTG, GABAergic.Inh_L1_SST_NMBR (181 genes)\n",
      "querying 1-181...done.\n",
      "Finished.\n",
      "1 input query terms found no hit:\n",
      "\t['ENSG00000170647']\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "(39/40) Converting Allen_human_MTG, GABAergic.Inh_L1-2_PAX6_CDH12 (181 genes)\n",
      "querying 1-181...done.\n",
      "Finished.\n",
      "1 input query terms found no hit:\n",
      "\t['ENSG00000170647']\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "(40/40) Converting tabula_muris, Marrow.precursor_B_cell (143 genes)\n",
      "querying 1-143...done.\n",
      "Finished.\n",
      "2 input query terms found no hit:\n",
      "\t['ENSG00000166748', 'ENSG00000197697']\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n",
      "\n",
      "Running Enrichr...\n",
      "\n",
      "(1/40) Analyzing tabula_muris, Brain_Non-Myeloid.brain_pericyte (143 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(2/40) Analyzing Descartes_Human_Cerebrum, Cerebrum-Vascular_endothelial_cells (152 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(3/40) Analyzing Descartes_Human_Cerebellum, Cerebellum-Vascular_endothelial_cells (71 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(4/40) Analyzing Descartes_Human_Cerebrum, Cerebrum-Excitatory_neurons (152 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(5/40) Analyzing tabula_muris, Heart.smooth_muscle_cell (143 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(6/40) Analyzing GSE67835_Human_Cortex, fetal_quiescent (39 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(7/40) Analyzing tabula_muris, Heart.myofibroblast_cell (143 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(8/40) Analyzing tabula_muris, Brain_Non-Myeloid.endothelial_cell (143 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(9/40) Analyzing GSE67835_Human_Cortex, endothelial (39 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(10/40) Analyzing Descartes_Human_Cerebrum, Cerebrum-Inhibitory_neurons (152 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(11/40) Analyzing Descartes_Human_Cerebellum, Cerebellum-Inhibitory_interneurons (71 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(12/40) Analyzing tabula_muris, Marrow.late_pro-B_cell (143 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(13/40) Analyzing PsychENCODE_DER-22, Ex8 (146 genes)...\n",
      "Writing to file...\n",
      "\n",
      "(14/40) Analyzing tabula_muris, Fat.mesenchymal_stem_cell_of_adipose (143 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(15/40) Analyzing tabula_muris, Marrow.granulocytopoietic_cell (143 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(16/40) Analyzing mousebrain, VECA (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(17/40) Analyzing mousebrain, SCINH5 (148 genes)...\n",
      "Writing to file...\n",
      "\n",
      "(18/40) Analyzing mousebrain, SCINH2 (147 genes)...\n",
      "Writing to file...\n",
      "\n",
      "(19/40) Analyzing mousebrain, SCINH10 (149 genes)...\n",
      "Writing to file...\n",
      "\n",
      "(20/40) Analyzing mousebrain, HBINH8 (149 genes)...\n",
      "Writing to file...\n",
      "\n",
      "(21/40) Analyzing mousebrain, ENTG1 (147 genes)...\n",
      "Writing to file...\n",
      "\n",
      "(22/40) Analyzing mousebrain, DEGLU4 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(23/40) Analyzing PsychENCODE_DER-22, Oligo (146 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(24/40) Analyzing Allen_human_LGN, GABAergic.LGN_Inh_LAMP5 (125 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(25/40) Analyzing PsychENCODE_DER-22, Ex6a (146 genes)...\n",
      "Writing to file...\n",
      "\n",
      "(26/40) Analyzing GSE67835_Human_Cortex, oligodendrocytes (39 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(27/40) Analyzing Allen_human_LGN, Glutamatergic.LGN_Exc_BTNL9 (125 genes)...\n",
      "Writing to file...\n",
      "\n",
      "(28/40) Analyzing GSE101601_Temporal_Cortex, htrc2 (16 genes)...\n",
      "Writing to file...\n",
      "\n",
      "(29/40) Analyzing DroNc_Human_Hippocampus, exPFC1 (124 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(30/40) Analyzing DroNc_Human_Hippocampus, exDG (124 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(31/40) Analyzing DroNc_Human_Hippocampus, exCA3 (124 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(32/40) Analyzing DroNc_Human_Hippocampus, MG (112 genes)...\n",
      "Writing to file...\n",
      "\n",
      "(33/40) Analyzing DroNc_Human_Hippocampus, END (124 genes)...\n",
      "Writing to file...\n",
      "\n",
      "(34/40) Analyzing Descartes_Human_Cerebrum, Cerebrum-Astrocytes (152 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(35/40) Analyzing Allen_human_MTG, Glutamatergic.Exc_L5-6_THEMIS_FGF10 (179 genes)...\n",
      "Writing to file...\n",
      "\n",
      "(36/40) Analyzing Allen_human_MTG, Glutamatergic.Exc_L5-6_THEMIS_DCSTAMP (179 genes)...\n",
      "Writing to file...\n",
      "\n",
      "(37/40) Analyzing Allen_human_MTG, Glutamatergic.Exc_L5-6_THEMIS_CRABP1 (179 genes)...\n",
      "Writing to file...\n",
      "\n",
      "(38/40) Analyzing Allen_human_MTG, GABAergic.Inh_L1_SST_NMBR (180 genes)...\n",
      "Writing to file...\n",
      "\n",
      "(39/40) Analyzing Allen_human_MTG, GABAergic.Inh_L1-2_PAX6_CDH12 (180 genes)...\n",
      "Writing to file...\n",
      "\n",
      "(40/40) Analyzing tabula_muris, Marrow.precursor_B_cell (141 genes)...\n",
      "Writing to file...\n",
      "Performing GSEA...\n",
      "\n",
      "Top None ranked cell-types (N=71) in Schizophrenia GWAS:\n",
      "1. Allen_human_MTG: Glutamatergic.Exc_L2_LAMP5_LTK (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU24 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU9 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU8 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU7 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU5 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU4 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU3 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU23 (N=1, freq=1.0)\n",
      "1. mousebrain: TEINH11 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU22 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU21 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU20 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU2 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU19 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU18 (N=1, freq=1.0)\n",
      "1. mousebrain: TEINH1 (N=1, freq=1.0)\n",
      "1. mousebrain: TEINH12 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU15 (N=1, freq=1.0)\n",
      "1. tabula_muris: Brain_Non-Myeloid.neuron (N=1, freq=1.0)\n",
      "1. tabula_muris: Pancreas.pancreatic_A_cell (N=1, freq=1.0)\n",
      "1. tabula_muris: Marrow.granulocyte_monocyte_progenitor_cell (N=1, freq=1.0)\n",
      "1. tabula_muris: Marrow.common_lymphoid_progenitor (N=1, freq=1.0)\n",
      "1. tabula_muris: Marrow.Slamf1-positive_multipotent_progenitor_cell (N=1, freq=1.0)\n",
      "1. tabula_muris: Brain_Non-Myeloid.oligodendrocyte_precursor_cell (N=1, freq=1.0)\n",
      "1. tabula_muris: Brain_Non-Myeloid.oligodendrocyte (N=1, freq=1.0)\n",
      "1. mousebrain: TEINH20 (N=1, freq=1.0)\n",
      "1. mousebrain: TEINH13 (N=1, freq=1.0)\n",
      "1. mousebrain: TEINH19 (N=1, freq=1.0)\n",
      "1. mousebrain: TEINH18 (N=1, freq=1.0)\n",
      "1. mousebrain: TEINH17 (N=1, freq=1.0)\n",
      "1. mousebrain: TEINH16 (N=1, freq=1.0)\n",
      "1. mousebrain: TEINH15 (N=1, freq=1.0)\n",
      "1. mousebrain: TEINH14 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU17 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU14 (N=1, freq=1.0)\n",
      "1. Descartes_Human_Cerebrum: Cerebrum-Excitatory_neurons (N=1, freq=1.0)\n",
      "1. PsychENCODE_DER-22: Ex1 (N=1, freq=1.0)\n",
      "1. mousebrain: DEGLU4 (N=1, freq=1.0)\n",
      "1. mousebrain: DEGLU3 (N=1, freq=1.0)\n",
      "1. mousebrain: DEGLU2 (N=1, freq=1.0)\n",
      "1. mousebrain: DEGLU1 (N=1, freq=1.0)\n",
      "1. PsychENCODE_DER-22: Ex9 (N=1, freq=1.0)\n",
      "1. PsychENCODE_DER-22: Ex3e (N=1, freq=1.0)\n",
      "1. GSE67835_Human_Cortex: neurons (N=1, freq=1.0)\n",
      "1. mousebrain: DEINH3 (N=1, freq=1.0)\n",
      "1. DroNc_Human_Hippocampus: exPFC2 (N=1, freq=1.0)\n",
      "1. DroNc_Human_Hippocampus: exPFC1 (N=1, freq=1.0)\n",
      "1. DroNc_Human_Hippocampus: exCA1 (N=1, freq=1.0)\n",
      "1. DroNc_Human_Hippocampus: GABA2 (N=1, freq=1.0)\n",
      "1. Descartes_Human_Cerebrum: Cerebrum-Limbic_system_neurons (N=1, freq=1.0)\n",
      "1. Descartes_Human_Cerebrum: Cerebrum-Inhibitory_neurons (N=1, freq=1.0)\n",
      "1. mousebrain: DEINH1 (N=1, freq=1.0)\n",
      "1. mousebrain: HBGLU5 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU13 (N=1, freq=1.0)\n",
      "1. mousebrain: MSN2 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU12 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU11 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU10 (N=1, freq=1.0)\n",
      "1. mousebrain: TEGLU1 (N=1, freq=1.0)\n",
      "1. mousebrain: MSN5 (N=1, freq=1.0)\n",
      "1. mousebrain: MSN3 (N=1, freq=1.0)\n",
      "1. mousebrain: MSN1 (N=1, freq=1.0)\n",
      "1. mousebrain: HBGLU6 (N=1, freq=1.0)\n",
      "1. mousebrain: MEINH2 (N=1, freq=1.0)\n",
      "1. mousebrain: MEINH10 (N=1, freq=1.0)\n",
      "1. mousebrain: MEGLU6 (N=1, freq=1.0)\n",
      "1. mousebrain: HBINH5 (N=1, freq=1.0)\n",
      "1. mousebrain: HBGLU9 (N=1, freq=1.0)\n",
      "1. mousebrain: HBGLU7 (N=1, freq=1.0)\n",
      "1. tabula_muris: Pancreas.type_B_pancreatic_cell (N=1, freq=1.0)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Converting Ensembl ID to Gene Symbol...\n",
      "\n",
      "(1/71) Converting Allen_human_MTG, Glutamatergic.Exc_L2_LAMP5_LTK (181 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(2/71) Converting mousebrain, TEGLU24 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(3/71) Converting mousebrain, TEGLU9 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(4/71) Converting mousebrain, TEGLU8 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(5/71) Converting mousebrain, TEGLU7 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(6/71) Converting mousebrain, TEGLU5 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(7/71) Converting mousebrain, TEGLU4 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(8/71) Converting mousebrain, TEGLU3 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(9/71) Converting mousebrain, TEGLU23 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(10/71) Converting mousebrain, TEINH11 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(11/71) Converting mousebrain, TEGLU22 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(12/71) Converting mousebrain, TEGLU21 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(13/71) Converting mousebrain, TEGLU20 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(14/71) Converting mousebrain, TEGLU2 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(15/71) Converting mousebrain, TEGLU19 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(16/71) Converting mousebrain, TEGLU18 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(17/71) Converting mousebrain, TEINH1 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(18/71) Converting mousebrain, TEINH12 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(19/71) Converting mousebrain, TEGLU15 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(20/71) Converting tabula_muris, Brain_Non-Myeloid.neuron (143 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(21/71) Converting tabula_muris, Pancreas.pancreatic_A_cell (143 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(22/71) Converting tabula_muris, Marrow.granulocyte_monocyte_progenitor_cell (143 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(23/71) Converting tabula_muris, Marrow.common_lymphoid_progenitor (143 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(24/71) Converting tabula_muris, Marrow.Slamf1-positive_multipotent_progenitor_cell (143 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(25/71) Converting tabula_muris, Brain_Non-Myeloid.oligodendrocyte_precursor_cell (143 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(26/71) Converting tabula_muris, Brain_Non-Myeloid.oligodendrocyte (143 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(27/71) Converting mousebrain, TEINH20 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(28/71) Converting mousebrain, TEINH13 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(29/71) Converting mousebrain, TEINH19 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(30/71) Converting mousebrain, TEINH18 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(31/71) Converting mousebrain, TEINH17 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(32/71) Converting mousebrain, TEINH16 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(33/71) Converting mousebrain, TEINH15 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(34/71) Converting mousebrain, TEINH14 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(35/71) Converting mousebrain, TEGLU17 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(36/71) Converting mousebrain, TEGLU14 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(37/71) Converting Descartes_Human_Cerebrum, Cerebrum-Excitatory_neurons (152 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(38/71) Converting PsychENCODE_DER-22, Ex1 (146 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(39/71) Converting mousebrain, DEGLU4 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(40/71) Converting mousebrain, DEGLU3 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(41/71) Converting mousebrain, DEGLU2 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(42/71) Converting mousebrain, DEGLU1 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(43/71) Converting PsychENCODE_DER-22, Ex9 (146 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(44/71) Converting PsychENCODE_DER-22, Ex3e (146 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(45/71) Converting GSE67835_Human_Cortex, neurons (39 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(46/71) Converting mousebrain, DEINH3 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(47/71) Converting DroNc_Human_Hippocampus, exPFC2 (124 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(48/71) Converting DroNc_Human_Hippocampus, exPFC1 (124 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(49/71) Converting DroNc_Human_Hippocampus, exCA1 (124 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(50/71) Converting DroNc_Human_Hippocampus, GABA2 (124 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(51/71) Converting Descartes_Human_Cerebrum, Cerebrum-Limbic_system_neurons (152 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(52/71) Converting Descartes_Human_Cerebrum, Cerebrum-Inhibitory_neurons (152 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(53/71) Converting mousebrain, DEINH1 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(54/71) Converting mousebrain, HBGLU5 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(55/71) Converting mousebrain, TEGLU13 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(56/71) Converting mousebrain, MSN2 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(57/71) Converting mousebrain, TEGLU12 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(58/71) Converting mousebrain, TEGLU11 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(59/71) Converting mousebrain, TEGLU10 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(60/71) Converting mousebrain, TEGLU1 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(61/71) Converting mousebrain, MSN5 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(62/71) Converting mousebrain, MSN3 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(63/71) Converting mousebrain, MSN1 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(64/71) Converting mousebrain, HBGLU6 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(65/71) Converting mousebrain, MEINH2 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(66/71) Converting mousebrain, MEINH10 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(67/71) Converting mousebrain, MEGLU6 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(68/71) Converting mousebrain, HBINH5 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(69/71) Converting mousebrain, HBGLU9 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(70/71) Converting mousebrain, HBGLU7 (151 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "(71/71) Converting tabula_muris, Pancreas.type_B_pancreatic_cell (143 genes)\n",
      "Cell-type already analyzed. Skipping conversion...\n",
      "\n",
      "Running Enrichr...\n",
      "\n",
      "(1/71) Analyzing Allen_human_MTG, Glutamatergic.Exc_L2_LAMP5_LTK (181 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(2/71) Analyzing mousebrain, TEGLU24 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(3/71) Analyzing mousebrain, TEGLU9 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(4/71) Analyzing mousebrain, TEGLU8 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "(5/71) Analyzing mousebrain, TEGLU7 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(6/71) Analyzing mousebrain, TEGLU5 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(7/71) Analyzing mousebrain, TEGLU4 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(8/71) Analyzing mousebrain, TEGLU3 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(9/71) Analyzing mousebrain, TEGLU23 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(10/71) Analyzing mousebrain, TEINH11 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(11/71) Analyzing mousebrain, TEGLU22 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(12/71) Analyzing mousebrain, TEGLU21 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(13/71) Analyzing mousebrain, TEGLU20 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(14/71) Analyzing mousebrain, TEGLU2 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(15/71) Analyzing mousebrain, TEGLU19 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(16/71) Analyzing mousebrain, TEGLU18 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(17/71) Analyzing mousebrain, TEINH1 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(18/71) Analyzing mousebrain, TEINH12 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(19/71) Analyzing mousebrain, TEGLU15 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(20/71) Analyzing tabula_muris, Brain_Non-Myeloid.neuron (143 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(21/71) Analyzing tabula_muris, Pancreas.pancreatic_A_cell (143 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(22/71) Analyzing tabula_muris, Marrow.granulocyte_monocyte_progenitor_cell (143 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(23/71) Analyzing tabula_muris, Marrow.common_lymphoid_progenitor (143 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(24/71) Analyzing tabula_muris, Marrow.Slamf1-positive_multipotent_progenitor_cell (143 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(25/71) Analyzing tabula_muris, Brain_Non-Myeloid.oligodendrocyte_precursor_cell (143 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(26/71) Analyzing tabula_muris, Brain_Non-Myeloid.oligodendrocyte (143 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(27/71) Analyzing mousebrain, TEINH20 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(28/71) Analyzing mousebrain, TEINH13 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(29/71) Analyzing mousebrain, TEINH19 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(30/71) Analyzing mousebrain, TEINH18 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(31/71) Analyzing mousebrain, TEINH17 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(32/71) Analyzing mousebrain, TEINH16 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(33/71) Analyzing mousebrain, TEINH15 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(34/71) Analyzing mousebrain, TEINH14 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(35/71) Analyzing mousebrain, TEGLU17 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(36/71) Analyzing mousebrain, TEGLU14 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(37/71) Analyzing Descartes_Human_Cerebrum, Cerebrum-Excitatory_neurons (152 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(38/71) Analyzing PsychENCODE_DER-22, Ex1 (146 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(39/71) Analyzing mousebrain, DEGLU4 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(40/71) Analyzing mousebrain, DEGLU3 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(41/71) Analyzing mousebrain, DEGLU2 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(42/71) Analyzing mousebrain, DEGLU1 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(43/71) Analyzing PsychENCODE_DER-22, Ex9 (146 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(44/71) Analyzing PsychENCODE_DER-22, Ex3e (146 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(45/71) Analyzing GSE67835_Human_Cortex, neurons (39 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(46/71) Analyzing mousebrain, DEINH3 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(47/71) Analyzing DroNc_Human_Hippocampus, exPFC2 (124 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(48/71) Analyzing DroNc_Human_Hippocampus, exPFC1 (124 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(49/71) Analyzing DroNc_Human_Hippocampus, exCA1 (124 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(50/71) Analyzing DroNc_Human_Hippocampus, GABA2 (124 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(51/71) Analyzing Descartes_Human_Cerebrum, Cerebrum-Limbic_system_neurons (152 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(52/71) Analyzing Descartes_Human_Cerebrum, Cerebrum-Inhibitory_neurons (152 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(53/71) Analyzing mousebrain, DEINH1 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(54/71) Analyzing mousebrain, HBGLU5 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(55/71) Analyzing mousebrain, TEGLU13 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(56/71) Analyzing mousebrain, MSN2 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(57/71) Analyzing mousebrain, TEGLU12 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(58/71) Analyzing mousebrain, TEGLU11 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(59/71) Analyzing mousebrain, TEGLU10 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(60/71) Analyzing mousebrain, TEGLU1 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(61/71) Analyzing mousebrain, MSN5 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(62/71) Analyzing mousebrain, MSN3 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(63/71) Analyzing mousebrain, MSN1 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(64/71) Analyzing mousebrain, HBGLU6 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(65/71) Analyzing mousebrain, MEINH2 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(66/71) Analyzing mousebrain, MEINH10 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(67/71) Analyzing mousebrain, MEGLU6 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(68/71) Analyzing mousebrain, HBINH5 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(69/71) Analyzing mousebrain, HBGLU9 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(70/71) Analyzing mousebrain, HBGLU7 (151 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n",
      "\n",
      "(71/71) Analyzing tabula_muris, Pancreas.type_B_pancreatic_cell (143 genes)...\n",
      "Cell-type already analyzed. Skipping analysis...\n"
     ]
    }
   ],
   "source": [
    "gwas_group_dict = {\n",
    "    'Structural_MRI':['volume','area','thickness'],\n",
    "    'DTI_tracts':['DTI'],\n",
    "    'rsfMRI':['Net'],\n",
    "    'Schizophrenia': ['SCZ_PGC3_2020']\n",
    "}\n",
    "for name,keywords in gwas_group_dict.items():\n",
    "    gsea_dict = gene_set_enrichment_analysis.gsea(df_all, {name:keywords}, rank=None) # rank=None, analyze all celltypes\n",
    "    gsea_summary_df = gene_set_enrichment_analysis.summarize_gsea(gsea_dict,\n",
    "                                                                  min_count=1, # don't save terms only found in 1 celltype \n",
    "                                                                  save_to_excel=True,\n",
    "                                                                  filename=f'gsea/summaries/gsea_{name}_summary.xlsx',    \n",
    "                                                             )"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## FIgure 9"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# genetic correlation on a subset of phenotypes was performed using ldsc\n",
    "genetic_correlation_dir = f\"{figure_dir}/genetic_correlation_out/\"\n",
    "columns = ['p1','p2','rg','se','z','p','h2_obs','h2_obs_se','h2_int','h2_int_se','gcov_int','gcov_int_se']\n",
    "data = []\n",
    "for file in Path(genetic_correlation_dir).iterdir():\n",
    "    with open(file,'r') as f:\n",
    "        lines = f.readlines()\n",
    "        for linenumber,line in enumerate(lines):\n",
    "            if line.startswith('p1'):\n",
    "                d = (lines[linenumber+1]).strip().split()\n",
    "                d[0] = d[0].split('/')[-1].split('.')[0]\n",
    "                d[1] = d[1].split('/')[-1].split('.')[0]\n",
    "                data.append(d)\n",
    "                break\n",
    "df = pd.DataFrame(data, columns=columns).replace('NA',np.nan)\n",
    "df = df.apply(pd.to_numeric, errors='ignore')\n",
    "display(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# the cell type correlation uses different names for the same phenotypes\n",
    "# so first we will convert these names so we can compare them\n",
    "phenotypes_cort = {\n",
    "                \"Frontal\": [\"superiorfrontal\", \"rostralmiddlefrontal\", \"caudalmiddlefrontal\", \"parsopercularis\", \"parstriangularis\",\"parsorbitalis\", \"lateralorbitofrontal\", \"medialorbitofrontal\", \"precentral\", \"paracentral\", \"frontalpole\"],\n",
    "                \"Temporal\": [\"superiortemporal\", \"middletemporal\", \"inferiortemporal\", \"bankssts\", \"fusiform\", \"transversetemporal\",\"entorhinal\", \"temporalpole\", \"parahippocampal\"],\n",
    "                \"Parietal\": [\"superiorparietal\", \"inferiorparietal\", \"supramarginal\", \"postcentral\", \"precuneus\"  ],\n",
    "                \"Occipital\": [\"lateraloccipital\", \"lingual\", \"cuneus\", \"pericalcarine\"],\n",
    "                \"Cingulate\": [\"rostralanteriorcingulate\", \"caudalanteriorcingulate\", \"posteriorcingulate\", \"isthmuscingulate\"],\n",
    "                \"Insula\": [\"insula\"],  \n",
    "                \"Cortex\": ['cortex']\n",
    "                }\n",
    "model_cort = {\n",
    "        \"model1a\": \"surface area\",\n",
    "        \"model1b\": \"cortical thickness (adjusted)\",\n",
    "        \"model1c\": \"cortical thickness\",\n",
    "        \"model1d\": \"volume\" \n",
    "        }\n",
    "inv_map_pheno = {v:k for k,v_list in phenotypes_cort.items() for v in v_list}\n",
    "p_df = pd.concat([df['p1'],df['p2']])\n",
    "df_cort = p_df[p_df.str.contains('model1\\w_\\w+')].copy()\n",
    "keys_cortical = df_cort.unique().tolist()\n",
    "df_cort = df_cort.str.extract('(model1\\w)_(\\w+)_\\w+')\n",
    "df_cort[0] = df_cort[0].apply(lambda x: model_cort[x])\n",
    "df_cort[2] = df_cort[1].apply(lambda x: inv_map_pheno[x])\n",
    "values_cortical = (df_cort[0]+': '+df_cort[2]+'.'+df_cort[1]).values.tolist()\n",
    "cortical_dict = {k:v for k,v in zip(keys_cortical,values_cortical)}\n",
    "\n",
    "df_dti = p_df[p_df.str.contains('dti|fapc')].copy()\n",
    "keys_dti = df_dti.unique().tolist()\n",
    "meta_dti = pd.read_csv('gwas/DTI.csv', usecols=[0,1])\n",
    "meta_dti[['type','pheno']] = meta_dti['assoc_files'].str.extract(r'_(\\w+)_(pheno\\d+)', expand=True)\n",
    "df_dti = df_dti.str.extract('ukb_phase1to3_(\\w+)_dec21_2019_(\\w+)').replace({'fapcs':'fpc','dti441':'mean'})\n",
    "df_dti = df_dti.merge(meta_dti, right_on=['type','pheno'], left_on=[0,1]).drop_duplicates()\n",
    "df_dti[['region','param']] = df_dti['phenocode'].str.split('_', expand=True)\n",
    "values_dti = ('DTI '+df_dti['param']+': '+df_dti['region']).values.tolist()\n",
    "dti_dict = {k:v for k,v in zip(keys_dti,values_dti)}\n",
    "\n",
    "df_subcor = p_df[p_df.str.contains('fixed_\\w+_MEGAMETA')].copy()\n",
    "keys_subcortical = df_subcor.unique().tolist()\n",
    "values_subcortical = ('volume: '+df_subcor.str.extract(r'fixed_(\\w+)_MEGAMETA')).values.flatten().tolist()\n",
    "subcortical_dict = {k:v for k,v in zip(keys_subcortical,values_subcortical)}\n",
    "\n",
    "hipp_dict = {'ukb_roi_volume_may12_2019_phase1and2_pheno81_allchr_withA2':'volume: Hippocampus.left',\n",
    "             'ukb_roi_volume_may12_2019_phase1and2_pheno94_allchr_withA2':'volume: Hippocampus.right'\n",
    "            }\n",
    "\n",
    "df_rsfmri = p_df[p_df.str.contains('fmri')].copy()\n",
    "keys_rsfmri = df_rsfmri.unique().tolist()\n",
    "rsfmri_metafile = \"gwas/fMRI.csv\"\n",
    "rsfmri_metadata = pd.read_csv(rsfmri_metafile)\n",
    "rsfmri_metadata[['type','pheno']] = rsfmri_metadata['assoc_files'].str.extract('fmri_(\\w+)_(pheno\\d+)', expand=True)\n",
    "df_rsfmri = df_rsfmri.str.extract('fmri_(\\w+)_dec21_2019_(pheno\\d+)')\n",
    "values_rsfmri = df_rsfmri.merge(rsfmri_metadata, left_on=[0,1], right_on=['type','pheno']).drop_duplicates()['phenocode'].values.tolist()\n",
    "rsfmri_dict = {k:v for k,v in zip(keys_rsfmri,values_rsfmri)}\n",
    "\n",
    "others_dict = {'ASD_iPSYCH-PGC_ASD_Nov2017':'ASD_2019',\n",
    "               'PGC_UKB_depression2019':'PGC_depression2019',\n",
    "               'SCZ_clozuk_pgc2':'SCZ_2016',\n",
    "               'CHARGE-ENIGMA-HV-METAANALYSIS':'HV_2016'    \n",
    "}\n",
    "\n",
    "merged_dict = {**cortical_dict,**dti_dict,**subcortical_dict,**rsfmri_dict,**hipp_dict,**others_dict}\n",
    "\n",
    "def replace_name(name):\n",
    "    if name in merged_dict.keys():\n",
    "        return merged_dict[name]\n",
    "    else:\n",
    "        return name\n",
    "    \n",
    "df['p1'] = df['p1'].apply(replace_name)\n",
    "df['p2'] = df['p2'].apply(replace_name)\n",
    "display(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# check which phenotype has not had their name changed\n",
    "for key in df['p1'].unique().tolist():\n",
    "    if key not in merged_dict.values():\n",
    "        print(key)\n",
    "# these phenotypes have the same name in the cell type correlation dataframe so no further renaming is needed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "gc_corr_sorted = df[['p1','p2','rg','p']].copy() # genetic correlation\n",
    "ct_corr_sorted = corr_df.copy() # cell type correlation\n",
    "\n",
    "# sort the p1/gwasx and p2/gwasy alphabetically so we can easily merge them\n",
    "gc_corr_sorted[['p1','p2']] = gc_corr_sorted.apply(\n",
    "    lambda x: sorted([x[0],x[1]]),\n",
    "    result_type='expand', axis=1\n",
    ")\n",
    "ct_corr_sorted[['gwasx','gwasy']] = ct_corr_sorted.apply(\n",
    "    lambda x: sorted([x[0],x[1]]),\n",
    "    result_type='expand', axis=1\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_corr = gc_corr_sorted.merge(ct_corr_sorted, right_on=['gwasx','gwasy'],\n",
    "                                   left_on=['p1','p2']).drop(columns=['p1','p2'])\n",
    "display(merged_corr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "plt.style.use('seaborn')\n",
    "fig, axes = plt.subplots(1,3,figsize=(15,8), sharey=True)\n",
    "s = 2\n",
    "alpha = .5\n",
    "color_map = {\n",
    "    'Not Significant' : 'black',\n",
    "    'Significant in\\nGenetic Correlation' : 'dodgerblue',\n",
    "    'Significant in\\nCell Type Correlation' : 'firebrick',\n",
    "    'Significant in Both' : 'seagreen'\n",
    "}\n",
    "for ax,m in zip(axes.flatten(),methods):\n",
    "    merged_corr_m = merged_corr[merged_corr['method']==m]\n",
    "    threshold = 0.05/merged_corr_m.shape[0]\n",
    "    not_significant = merged_corr_m[(merged_corr_m['p']>threshold)& # gc not sign and\n",
    "                                    ((merged_corr_m['pval']>threshold) # ct not sign\n",
    "                                     |(merged_corr_m['corr']<0))] # or ct corr < 0\n",
    "    gc_significant = merged_corr_m[(merged_corr_m['p']<=threshold)& # gc sign and\n",
    "                                   ((merged_corr_m['pval']>threshold) # ct not sign\n",
    "                                     |(merged_corr_m['corr']<0))] # or ct corr < 0\n",
    "    ct_significant = merged_corr_m[(merged_corr_m['p']>threshold)& # gc not sign and\n",
    "                                   ((merged_corr_m['pval']<=threshold)& # ct sign and\n",
    "                                    (merged_corr_m['corr']>0))] # ct corr >0 \n",
    "    both_significant = merged_corr_m[(merged_corr_m['p']<=threshold)& # gc sign and\n",
    "                                     ((merged_corr_m['pval']<=threshold)& # ct sign and\n",
    "                                      (merged_corr_m['corr']>0))] # ct corr >0     \n",
    "    ax.scatter(not_significant['corr'],not_significant['rg'], color=color_map['Not Significant'],\n",
    "               alpha=alpha, s=s)\n",
    "    ax.scatter(gc_significant['corr'],gc_significant['rg'], color=color_map['Significant in\\nGenetic Correlation'],\n",
    "               alpha=alpha, s=s)\n",
    "    ax.scatter(ct_significant['corr'],ct_significant['rg'], color=color_map['Significant in\\nCell Type Correlation'],\n",
    "               alpha=alpha, s=s)\n",
    "    ax.scatter(both_significant['corr'],both_significant['rg'], color=color_map['Significant in Both'],\n",
    "               alpha=alpha, s=s)\n",
    "    \n",
    "    ax.set_xlim(-1.1,1.1)\n",
    "    ax.set_title(m)\n",
    "    ax.axhline(0, color='w', lw=3, ls='-', zorder=0)\n",
    "    ax.axvline(0, color='w', lw=3, ls='-', zorder=0)\n",
    "#     ax.grid()\n",
    "fig.text(0.5, 0.04, 'Cell Type Correlation', ha='center')\n",
    "fig.text(0.08, 0.5, 'Genetic Correlation', va='center', rotation='vertical')\n",
    "patches = [mpatches.Patch(color=v, label=k) for k,v in color_map.items()]\n",
    "fig.legend(handles=patches, bbox_to_anchor=(1.04, .89), loc=1, frameon=False)\n",
    "fig.savefig(f\"{figure_dir}genetic_corr_vs_celltype_corr.png\", dpi=200, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.style.use('seaborn')\n",
    "fig, axes = plt.subplots(1,3,figsize=(15,8), sharey=True)\n",
    "s = 2\n",
    "alpha = .5\n",
    "color_map = {\n",
    "    'Not Significant' : 'black',\n",
    "    'Significant in\\nGenetic Correlation' : 'dodgerblue',\n",
    "    'Significant in\\nCell Type Correlation' : 'firebrick',\n",
    "    'Significant in Both' : 'seagreen'\n",
    "}\n",
    "ct_thres = corr_df[(corr_df['corr']<0)]['pval'].min()\n",
    "for ax,m in zip(axes.flatten(),methods):\n",
    "    merged_corr_m = merged_corr[merged_corr['method']==m]\n",
    "    threshold = 0.05/merged_corr_m.shape[0]\n",
    "    not_significant = merged_corr_m[(merged_corr_m['p']>threshold)& # gc not sign and\n",
    "                                    (merged_corr_m['pval']>ct_thres)] # ct not sign\n",
    "    gc_significant = merged_corr_m[(merged_corr_m['p']<=threshold)& # gc sign and\n",
    "                                   (merged_corr_m['pval']>ct_thres)] # ct not sign\n",
    "    ct_significant = merged_corr_m[(merged_corr_m['p']>threshold)& # gc not sign and\n",
    "                                   (merged_corr_m['pval']<=ct_thres)] # ct sign \n",
    "    both_significant = merged_corr_m[(merged_corr_m['p']<=threshold)& # gc sign and\n",
    "                                     (merged_corr_m['pval']<=ct_thres)] # ct sign\n",
    "    ax.scatter(not_significant['corr'],not_significant['rg'], color=color_map['Not Significant'],\n",
    "               alpha=alpha, s=s)\n",
    "    ax.scatter(gc_significant['corr'],gc_significant['rg'], color=color_map['Significant in\\nGenetic Correlation'],\n",
    "               alpha=alpha, s=s)\n",
    "    ax.scatter(ct_significant['corr'],ct_significant['rg'], color=color_map['Significant in\\nCell Type Correlation'],\n",
    "               alpha=alpha, s=s)\n",
    "    ax.scatter(both_significant['corr'],both_significant['rg'], color=color_map['Significant in Both'],\n",
    "               alpha=alpha, s=s)\n",
    "    \n",
    "    ax.set_xlim(-1.1,1.1)\n",
    "    ax.set_title(m)\n",
    "    ax.axhline(0, color='w', lw=3, ls='-', zorder=0)\n",
    "    ax.axvline(0, color='w', lw=3, ls='-', zorder=0)\n",
    "#     ax.grid()\n",
    "fig.text(0.5, 0.04, 'Cell Type Correlation', ha='center')\n",
    "fig.text(0.08, 0.5, 'Genetic Correlation', va='center', rotation='vertical')\n",
    "patches = [mpatches.Patch(color=v, label=k) for k,v in color_map.items()]\n",
    "fig.legend(handles=patches, bbox_to_anchor=(1.04, .89), loc=1, frameon=False)\n",
    "fig.savefig(f\"{figure_dir}genetic_corr_vs_celltype_corr_other_pthresh.png\", dpi=200, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Figure 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "mri_corr_df = corr_df[(corr_df['gwasx'].str.contains(': \\w+\\.\\w+'))& # only get MRI measurement phenotypes\n",
    "                          (corr_df['gwasy'].str.contains(': \\w+\\.\\w+'))&\n",
    "                          ~((corr_df['gwasx'].str.contains('Hippocampus'))| # remove hippocampus volume data\n",
    "                           corr_df['gwasy'].str.contains('Hippocampus'))\n",
    "                         ] \n",
    "all_mri_gwas = pd.concat([mri_corr_df['gwasx'],mri_corr_df['gwasy']]).unique()\n",
    "diag_df = [[g,g,1,0,m] for g in all_mri_gwas for m in methods]\n",
    "mri_corr_df = mri_corr_df.append(pd.DataFrame(diag_df, columns=mri_corr_df.columns))\n",
    "mri_corr_df = mri_corr_df[mri_corr_df['method']==m]\n",
    "mri_corr_df = mri_corr_df.pivot(columns='gwasy', index='gwasx', values='corr')\n",
    "mri_corr_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.style.use('default')\n",
    "fig, ax = plt.subplots(1,1, figsize=(5,5))\n",
    "for m in ['H-MAGMA']:\n",
    "    mri_corr_df = corr_df[(corr_df['gwasx'].str.contains(': \\w+\\.\\w+'))& # only get MRI measurement phenotypes\n",
    "                          (corr_df['gwasy'].str.contains(': \\w+\\.\\w+'))&\n",
    "                          ~((corr_df['gwasx'].str.contains('Hippocampus'))| # remove hippocampus volume data\n",
    "                           corr_df['gwasy'].str.contains('Hippocampus'))\n",
    "                         ] \n",
    "    all_mri_gwas = pd.concat([mri_corr_df['gwasx'],mri_corr_df['gwasy']]).unique()\n",
    "    diag_df = [[g,g,1,0,m] for g in all_mri_gwas for m in methods]\n",
    "    mri_corr_df = mri_corr_df.append(pd.DataFrame(diag_df, columns=mri_corr_df.columns))\n",
    "    mri_corr_df = mri_corr_df[mri_corr_df['method']==m]\n",
    "    mri_corr_df = mri_corr_df.pivot(columns='gwasy', index='gwasx', values='corr')\n",
    "    ax.imshow(mri_corr_df,\n",
    "              cmap='bwr', vmin=-1, vmax=1)\n",
    "    ax.set_title(m)\n",
    "    ax.tick_params(axis=u'both', which=u'both',length=0)\n",
    "    ax.spines['top'].set_visible(False)\n",
    "    ax.spines['right'].set_visible(False)\n",
    "    ax.spines['bottom'].set_visible(False)\n",
    "    ax.spines['left'].set_visible(False)\n",
    "##     yticks = mri_corr_df.index.str.split(':').str[0].str.split(' ').str[-1].to_list()\n",
    "    yticks = mri_corr_df.index.str.split(':').str[0].str.split('(').str[0].str.rstrip().str.split(' ').str[-1]\n",
    "    ticks = ['']*len(yticks)\n",
    "    for group in set(yticks):\n",
    "        start = False\n",
    "        end = False\n",
    "        for i,tick in enumerate(yticks):\n",
    "            if tick==group and start==False:\n",
    "                start = i\n",
    "            if start and tick!=group:\n",
    "                end = i\n",
    "                break\n",
    "        if end == False:\n",
    "            end = i\n",
    "#         ax.axvline(end, color='k', lw=0.5)\n",
    "        ax.axhline(end-.5, color='k', lw=0.5)\n",
    "        mid = int((end+start)/2)   \n",
    "    #     print(group, start, end, mid)\n",
    "        ticks[mid] = group\n",
    "    ax.axvline(35-.5, color='k', lw=0.5)\n",
    "    ax.axvline(70-.5, color='k', lw=0.5)\n",
    "    ax.yaxis.tick_right()\n",
    "    ax.set_yticks(list(range(0,mri_corr_df.shape[0])))\n",
    "    ax.set_yticklabels(ticks)\n",
    "    ax.xaxis.tick_top()\n",
    "    ax.set_xticks(list(range(0,mri_corr_df.shape[0])))\n",
    "    ax.set_xticklabels(ticks)\n",
    "#     break\n",
    "    plt.show()\n",
    "# plt.tight_layout()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "from statsmodels.stats.multitest import multipletests\n",
    "import numpy as np\n",
    "\n",
    "temp = corr_df.copy()\n",
    "# display(temp)\n",
    "temp['pval'] = multipletests(temp['pval'].values, method='bonferroni')[1]\n",
    "temp = temp[(temp['pval']<=0.05)\n",
    "     &(temp['corr']<0)\n",
    "    ].sort_values('corr').groupby(['gwasx','gwasy']).agg(list)\n",
    "\n",
    "temp['N_method'] = temp['method'].apply(lambda row: len(row))\n",
    "temp['mean_corr'] = temp['corr'].apply(lambda row: np.mean(row))\n",
    "display(temp[temp['N_method']>2].reset_index().drop(columns='pval').sort_values('mean_corr'))\n",
    "# temp[temp['corr'] < 0] = 0\n",
    "# temp.hist(bins=100)\n",
    "\n",
    "temp_list = temp[temp['N_method']>2].reset_index().drop(columns='pval').sort_values('mean_corr')[['gwasx','gwasy']].values.tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def make_name(string):\n",
    "    return string.replace(':','_')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "# scatter plot of GWAS 'a' and with all other significant correlated GWASes ('b')\n",
    "from adjustText import adjust_text\n",
    "methods = constants.METHODS\n",
    "method_color = {'H-MAGMA':'#4A148C',\n",
    "                'LDSC':'#CE93D8',\n",
    "                'MAGMA':'#9C27B0'}\n",
    "pval_correction = constants.PVAL_CORRECTION\n",
    "plot_annots = True\n",
    "plt.style.use('seaborn')\n",
    "tot_i = len(temp_list)*len(methods)\n",
    "i = 1\n",
    "for a,b in temp_list:\n",
    "    for m in methods:\n",
    "        print(f\"{i}/{tot_i}\")\n",
    "        i += 1\n",
    "        plot_name = f'{figure_dir}scatter_plots/{make_name(a)}_{make_name(b)}_{m}.png'\n",
    "        if not os.path.exists(plot_name):\n",
    "            fig, ax = plt.subplots(figsize=(7,7))\n",
    "            cy = 'blue'\n",
    "            cx = 'red'\n",
    "            ax.axhline(0, color='white', zorder=0, lw=3)\n",
    "            ax.axvline(0, color='white', zorder=0, lw=3)\n",
    "    #         ax.axline(xy1=(0,0), slope=-1, color='white', zorder=0, lw=3)\n",
    "            x,y = [df_all[(df_all.gwas==j)&(df_all.method==m).copy()]\n",
    "                   for j in [a,b]]\n",
    "            ax.scatter(x.beta, y.beta, label=m, color=method_color[m], alpha=.4)\n",
    "\n",
    "            x = x.assign(sign = lambda x: (x[f'pvalue_{pval_correction}'] <= 0.05))\n",
    "            y = y.assign(sign = lambda x: (x[f'pvalue_{pval_correction}'] <= 0.05))\n",
    "            #             x['sign'] = (x[f'pvalue_{pval_correction}'] <= 0.05)\n",
    "            #             y['sign'] = (y[f'pvalue_{pval_correction}'] <= 0.05)\n",
    "\n",
    "            # is really slow because of the annotations\n",
    "            if plot_annots:\n",
    "                texts_xy = [ax.annotate(annot, (x.beta.iat[l], y.beta.iat[l]),color='green',fontsize=8)\n",
    "                         for l,annot in enumerate(x.annotation) if x.sign.iat[l] and y.sign.iat[l]]\n",
    "                texts_x = [ax.annotate(annot, (x.beta.iat[l], y.beta.iat[l]),color=cx,fontsize=8)\n",
    "                         for l,annot in enumerate(x.annotation) if x.sign.iat[l] and not y.sign.iat[l] ]\n",
    "                texts_y = [ax.annotate(annot, (x.beta.iat[l], y.beta.iat[l]),color=cy,fontsize=8)\n",
    "                         for l,annot in enumerate(y.annotation) if y.sign.iat[l] and not x.sign.iat[l] ]\n",
    "                adjust_text(texts_xy+texts_x+texts_y, lim=100, arrowprops=dict(arrowstyle='->', color='black'))\n",
    "\n",
    "            # print(f'Plot {i+1}/{len(b_list)}...')\n",
    "            plt.title(m)\n",
    "\n",
    "            ax.set_xlabel(a, fontsize=10, color=cx)\n",
    "            ax.set_ylabel(b, fontsize=10, color=cy)\n",
    "\n",
    "            corr_val = corr_df[(((corr_df['gwasx']==a)&(corr_df['gwasy']==b))\n",
    "                               |\n",
    "                               ((corr_df['gwasx']==b)&(corr_df['gwasy']==a)))\n",
    "                               &\n",
    "                               (corr_df['method']==m)\n",
    "                              ]['corr'].values\n",
    "            ax.text(0.12, 0.97, f'Cell-type Correlation  = {float(corr_val):.2f}', horizontalalignment='center',\n",
    "                 verticalalignment='center', transform = ax.transAxes, size=7)\n",
    "            plt.tight_layout(rect=[0, 0.03, 1, 0.95])\n",
    "            min_value = min(min(x.beta),min(y.beta)) * 1.1\n",
    "            max_value = max(max(x.beta),max(y.beta)) * 1.1\n",
    "            ax.set_ylim(min_value,max_value)\n",
    "            ax.set_xlim(min_value,max_value)\n",
    "            # Path(f\"{out_dir}correlation_{make_filename(a)}\").mkdir(parents=True, exist_ok=True)\n",
    "            plt.savefig(plot_name,\n",
    "                        dpi=200, bbox_inches='tight')\n",
    "    #         plt.show()\n",
    "            plt.close(fig)\n",
    "    #         break\n",
    "    #     break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for gwas_group,keywords in gwas_group_dict.items():\n",
    "    regex_keywords = '|'.join(keywords)\n",
    "    df_group = df_all[df_all['gwas'].str.contains(regex_keywords)]\n",
    "    df_group_sign = df_group[(df_group[f\"pvalue_{pval_correction}\"]<0.05)]\n",
    "    display(df_group_sign)\n",
    "    break\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
